{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "ca6b7227-3add-487d-9595-8279c2728aa1",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(Seurat)\n",
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(ggpubr)\n",
    "\n",
    "library(dplyr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "934a2b1c-12e6-4401-b9ff-b40d4f1fe769",
   "metadata": {},
   "outputs": [],
   "source": [
    "list(\n",
    "'RPC (PC FGF17)' = c(\n",
    "    'FGF17',\n",
    "    'RBP1',\n",
    "    'DLK1',\n",
    "    'ZIC4',\n",
    "    'SP8',\n",
    "    'DUSP6',\n",
    "    'IL17RD',\n",
    "    'FABP7',\n",
    "    'FGF18',\n",
    "    'SPRY1',\n",
    "    'NFIA',\n",
    "    'PTN',\n",
    "    'ZIC2',\n",
    "    'RPL4',\n",
    "    'ZIC1',\n",
    "    'GALP',\n",
    "    'FDX1',\n",
    "    'FGF8',\n",
    "    'ETV4',\n",
    "    'FGF3',\n",
    "    'ZIC3',\n",
    "    'FEZF2',\n",
    "    'ZIC5',\n",
    "    'LYN',\n",
    "    'RSPO1',\n",
    "    'TGIF1',\n",
    "    'NFIB',\n",
    "    'SIX3',\n",
    "    'DIO3',\n",
    "    'ETV5',\n",
    "    'SPRY2',\n",
    "    'LHX2',\n",
    "    'DMRTA1',\n",
    "    'GLI3',\n",
    "    'ANXA2',\n",
    "    'EMX1'\n",
    ") , \n",
    "'AV NKX2-1 NKX6-2' = c(\n",
    "    'NKX6-2',\n",
    "    'FOXG1',\n",
    "    'SPON1',\n",
    "    'DIO3',\n",
    "    'FST',\n",
    "    'NDN',\n",
    "    'VAX1',\n",
    "    'LHX2',\n",
    "    'MAP1B',\n",
    "    'SIX3',\n",
    "    'MBIP',\n",
    "    'FZD5',\n",
    "    'MEST',\n",
    "    'EFNA5',\n",
    "    'PEG10',\n",
    "    'SOX3',\n",
    "    'NKX2-1',\n",
    "    'FEZ1',\n",
    "    'LMO2',\n",
    "    'FZD2',\n",
    "    'FOXO1',\n",
    "    'IGFBP5',\n",
    "    'TLE5',\n",
    "    'HES1',\n",
    "    'SHH',\n",
    "    'SOX1',\n",
    "    'SLC2A1',\n",
    "    'DNAJC1',\n",
    "    'MEIS3',\n",
    "    'TMEM132A',\n",
    "    'SYT1',\n",
    "    'FGFBP3',\n",
    "    'SHB',\n",
    "    'DLK1',\n",
    "    'PMP22',\n",
    "    'FRZB',\n",
    "    'LMO3',\n",
    "    'RSRP1',\n",
    "    'FZD8'\n",
    "), \n",
    "'AV NKX2-1 LMO1' = c(\n",
    "    'NNAT',\n",
    "    'DLK1',\n",
    "    'FRZB',\n",
    "    'LMO1',\n",
    "    'RBP1',\n",
    "    'NKX2-1',\n",
    "    'MBIP',\n",
    "    'FOXG1',\n",
    "    'OLIG1',\n",
    "    'FABP7',\n",
    "    'FGFBP3',\n",
    "    'HES5',\n",
    "    'OLIG2',\n",
    "    'NFIB',\n",
    "    'SOX6',\n",
    "    'DLX1',\n",
    "    'CCND1',\n",
    "    'DLX2',\n",
    "    'LAMP3',\n",
    "    'HEY1',\n",
    "    'ASCL1',\n",
    "    'HMGA1',\n",
    "    'PTPRZ1',\n",
    "    'HES4',\n",
    "    'FZD5',\n",
    "    'TCF4',\n",
    "    'PLP1',\n",
    "    'PCDH17',\n",
    "    'COL9A1',\n",
    "    'HMGA2',\n",
    "    'CXXC4',\n",
    "    'SIX3',\n",
    "    'SOX1',\n",
    "    'DUSP6',\n",
    "    'NOTCH1',\n",
    "    'SOX21',\n",
    "    'NES',\n",
    "    'AUTS2'\n",
    "), \n",
    "'GE RG NKX2-1 DLK1' = c(\n",
    "    'MEIS2',\n",
    "    'FOXG1',\n",
    "    'SFRP1',\n",
    "    'POU3F3',\n",
    "    'TSPAN7',\n",
    "    'FAM181B',\n",
    "    'LMO1',\n",
    "    'FABP5',\n",
    "    'FSTL5',\n",
    "    'STMN2',\n",
    "    'RBP1',\n",
    "    'H2AFX',\n",
    "    'DLK1',\n",
    "    'HMGA1',\n",
    "    'HOPX',\n",
    "    'FEZ1',\n",
    "    'MBIP',\n",
    "    'LAMP3',\n",
    "    'PBX3',\n",
    "    'NKX2-1',\n",
    "    'SOX1',\n",
    "    'HMGA2',\n",
    "    'FGFR1',\n",
    "    'FZD8',\n",
    "    'EFNB1',\n",
    "    'SOX6',\n",
    "    'CREB5',\n",
    "    'OLIG2',\n",
    "    'HDAC1',\n",
    "    'IDI1',\n",
    "    'PBX1'\n",
    "), \n",
    "'GE RG NKX2-1 OLIG1' = c(\n",
    "    'LMO1',\n",
    "    'ENHO',\n",
    "    'RBP1',\n",
    "    'OLIG2',\n",
    "    'SFRP1',\n",
    "    'FOXG1',\n",
    "    'NKX2-1',\n",
    "    'CDO1',\n",
    "    'POU3F3',\n",
    "    'MEIS2',\n",
    "    'HES5',\n",
    "    'CCND2',\n",
    "    'ASCL1',\n",
    "    'HES6',\n",
    "    'DLX1',\n",
    "    'HMGA1',\n",
    "    'NASP',\n",
    "    'OLIG1',\n",
    "    'SRSF2',\n",
    "    'STMN2',\n",
    "    'PBX3',\n",
    "    'DLX2',\n",
    "    'GSX2',\n",
    "    'PTPRZ1',\n",
    "    'HMGA2',\n",
    "    'FABP5',\n",
    "    'FGFBP3',\n",
    "    'RTN3',\n",
    "    'CNTFR',\n",
    "    'TCF4',\n",
    "    'SOX6',\n",
    "    'PCDH10',\n",
    "    'SOX8',\n",
    "    'LAMP3',\n",
    "    'MAP2',\n",
    "    'MBIP',\n",
    "    'NOTCH1',\n",
    "    'ZEB1',\n",
    "    'TFDP2',\n",
    "    'SOX1',\n",
    "    'TLE3',\n",
    "    'ZEB2',\n",
    "    'PSD3',\n",
    "    'BCL11B',\n",
    "    'IGF2BP1',\n",
    "    'CAMK2N1'\n",
    "), \n",
    "'Hem (PC RSPO3)' = c(\n",
    "    'RSPO3',    \n",
    "    'WLS',  \n",
    "    'C1H1orf61',    \n",
    "    'EMX2', \n",
    "    'IFT57',    \n",
    "    'RSPO2',    \n",
    "    'RSPO1',    \n",
    "    'HES1', \n",
    "    'DMRTA2',   \n",
    "    'WNT8B',    \n",
    "    'FEZF2',    \n",
    "    'MASP1',    \n",
    "    'TTYH1',    \n",
    "    'FGF13',    \n",
    "    'SP5',  \n",
    "    'TLE4', \n",
    "    'PEG3', \n",
    "    'HEY2', \n",
    "    'TCF4', \n",
    "    'TMTC4',    \n",
    "    'FGFR3',    \n",
    "    'DMRTA1',   \n",
    "    'NR2F1',    \n",
    "    'SULF2',    \n",
    "    'ID3',  \n",
    "    'ARX',  \n",
    "    'TGIF1',    \n",
    "    'EMX1', \n",
    "    'DLL1', \n",
    "    'TSHZ2',    \n",
    "    'DMRT3',    \n",
    "    'ID1',  \n",
    "    'WNT2B',    \n",
    "    'LHX9', \n",
    "    'AXIN2',    \n",
    "    'RPL37',    \n",
    "    'ZNF503',   \n",
    "    'RFX4', \n",
    "    'FSTL5',    \n",
    "    'CALB1',    \n",
    "    'NFIX', \n",
    "    'RORA', \n",
    "    'TMEM91',   \n",
    "    'HES4', \n",
    "    'BAMBI'\n",
    "), \n",
    "'Hem CPe (PC TTR)' = c(\n",
    "    'TTR',\n",
    "    'IGFBP7',\n",
    "    'PCP4',\n",
    "    'CLIC6',\n",
    "    'RWDD4',\n",
    "    'C13H2orf40',\n",
    "    'PCSK1N',\n",
    "    'ANXA2',\n",
    "    'HSPB1',\n",
    "    'C6H5orf49',\n",
    "    'CA2',\n",
    "    'FGGY',\n",
    "    'ATP11C',\n",
    "    'KCNJ13',\n",
    "    'RDX',\n",
    "    'TMEM88',\n",
    "    'ASB13',\n",
    "    'ROMO1',\n",
    "    'ID1',\n",
    "    'SULF1',\n",
    "    'ID3',\n",
    "    'DKK2',\n",
    "    'ID2',\n",
    "    'OTX2',\n",
    "    'SLC5A3',\n",
    "    'SLC29A4',\n",
    "    'SLC38A11',\n",
    "    'FOXJ1',\n",
    "    'PLTP',\n",
    "    'APOE',\n",
    "    'PDGFC',\n",
    "    'CRABP1',\n",
    "    'MAF',\n",
    "    'SERPINF1',\n",
    "    'WLS',\n",
    "    'H2AFJ',\n",
    "    'TMEM14A',\n",
    "    'PARP8'\n",
    "),\n",
    "'Antihem (PC SFRP2)' = c(\n",
    "    'PAX6',\n",
    "    'NR2F2',\n",
    "    'NR2F1',\n",
    "    'PTN',\n",
    "    'WNT7B',\n",
    "    'SFRP2',\n",
    "    'ATXN10',\n",
    "    'PTPRZ1',\n",
    "    'CDO1',\n",
    "    'SNCA',\n",
    "    'SLF1',\n",
    "    'NTRK2',\n",
    "    'FABP7',\n",
    "    'FEZF1',\n",
    "    'PTX3',\n",
    "    'LMO4',\n",
    "    'ARX',\n",
    "    'FAM181A',\n",
    "    'PENK',\n",
    "    'ID4',\n",
    "    'JUND',\n",
    "    'FGFR3',\n",
    "    'SOX21',\n",
    "    'FAM107A',\n",
    "    'DKK3'\n",
    "    ),\n",
    "\n",
    "'ZLI (PC TCFL2)' = c(\n",
    "    'NTS',\n",
    "    'APCDD1',\n",
    "    'WNT4',\n",
    "    'GRID2',\n",
    "    'CNTNAP2',\n",
    "    'CLYBL',\n",
    "    'BASP1',\n",
    "    'IGFBPL1',\n",
    "    'TCF7L2',\n",
    "    'BARHL2',\n",
    "    'IRX3',\n",
    "    'FST',\n",
    "    'PDZRN4',\n",
    "    'SHOX2',\n",
    "    'NTNG1',\n",
    "    'WNT3',\n",
    "    'WLS',\n",
    "    'FZD10',\n",
    "    'TIFA',\n",
    "    'NEUROG2',\n",
    "    'OTX1',\n",
    "    'PTN',\n",
    "    'POU3F2',\n",
    "    'RAB34',\n",
    "    'SDC2',\n",
    "    'IRX5',\n",
    "    'ID3',\n",
    "    'IRX2',\n",
    "    'FABP5',\n",
    "    'KCNQ2',\n",
    "    'SOCS1',\n",
    "    'SORCS2',\n",
    "    'NR2F1'\n",
    ")\n",
    ") -> nicola.selection2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3dfcf9aa-5322-4abf-933d-248ab425475a",
   "metadata": {},
   "outputs": [],
   "source": [
    "list(\n",
    "'RPC (PC FGF17)' = c(\n",
    "    'FABP7',\n",
    "    'GALP',\n",
    "    'PTN',\n",
    "    'DLK1',\n",
    "    'SIX3',\n",
    "    'SPRY1',\n",
    "    'RBP1',\n",
    "    'LHX2',\n",
    "    'DUSP6',\n",
    "    'ZIC1',\n",
    "    'FGF8',\n",
    "    'SP8'\n",
    ") , \n",
    "'AV' = c(\n",
    "    'MEIS3',\n",
    "    'IGFBP5',\n",
    "    'FZD5',\n",
    "    'EFNA5',\n",
    "    'FOXG1',\n",
    "    'FZD8',\n",
    "    'SOX1',\n",
    "    'TLE5',\n",
    "\n",
    "    'CCND1',\n",
    "    'HMGA2',\n",
    "    'HES5',\n",
    "    'SOX21',\n",
    "    'PLP1',\n",
    "    'SOX1',\n",
    "    'AUTS2'\n",
    "), \n",
    "'GE' = c(\n",
    "    'PBX1',\n",
    "    'SFRP1',\n",
    "    'MEIS2',\n",
    "    'FZD8',\n",
    "    'IDI1',\n",
    "\n",
    "    'ZEB1'\n",
    "), \n",
    "'Hem (PC RSPO3)' = c(\n",
    "    'HES1', \n",
    "    'TSHZ2',    \n",
    "    'FGF13',    \n",
    "    'TTYH1',    \n",
    "    'HES4', \n",
    "    'EMX2', \n",
    "    'ARX',  \n",
    "    'WLS',  \n",
    "    'DMRTA1',   \n",
    "    'WNT2B',    \n",
    "    'FEZF2',    \n",
    "    'TGIF1',    \n",
    "    'DLL1', \n",
    "    'FGFR3',    \n",
    "    'TCF4', \n",
    "    'RSPO2'\n",
    "), \n",
    "'Hem CPe (PC TTR)' = c(\n",
    "    'ID2',\n",
    "    'KCNJ13',\n",
    "    'HSPB1',\n",
    "    'ROMO1',\n",
    "    'RDX',\n",
    "    'OTX2',\n",
    "    'PLTP',\n",
    "    'SERPINF1',\n",
    "    'PARP8',\n",
    "    'SULF1'\n",
    "),\n",
    "'Antihem (PC SFRP2)' = c(\n",
    "    'JUND',\n",
    "    'FEZF1',\n",
    "    'WNT7B',\n",
    "    'PAX6',\n",
    "    'NR2F2',\n",
    "    'ID4',\n",
    "    'LMO4'\n",
    "),\n",
    "\n",
    "'ZLI (PC TCFL2)' = c(\n",
    "    'POU3F2',\n",
    "    'OTX1',\n",
    "    'IRX2',\n",
    "    'IRX3',\n",
    "    'BASP1',\n",
    "    'TCF7L2',\n",
    "    'SORCS2')\n",
    ") %>% lapply(unique) -> nicola.selection3 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "95b028ae-0553-4f80-9570-1fd1b6af19a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "nicola.selection4 <- list(\n",
    "    'FR' = c(\n",
    "        'MAP2', 'PAX6', 'WNT7B', 'POU3F2', 'HES4', 'FZD3', 'ALKBH7', 'HES5', 'ZIC1', 'DUSP6', 'FOXG1', 'NRG1', 'CXXC4', 'EFNA5', 'RBP1', 'ID2', 'SPRY1', 'EIF5A', 'PTN', 'IGFBP5', 'HMGA1', 'TPM1', 'SFRP1', 'FABP7', 'MEIS3', 'JUND'\n",
    "    )%>%rev(),\n",
    "    'MS' = c(\n",
    "        'LAMTOR3','RAB7A','SOX3','FGFR2','EMX2','IST1','NMT1','AP2B1','FZD8','FGFBP3','RAB11B','SRSF6' \n",
    "    )%>%rev(),\n",
    "    'OC' = c(\n",
    "        'FGFR3','ZNF219','ZBTB16','NR2F2','NR2F1','MAD2L2','IGDCC3','HES5','RPS27A','FEZF1','NELL2','SYT1','SNRPA','TECR','MGST1'\n",
    "    )%>%rev(),\n",
    "    'Tem' = c(\n",
    "        'FGFR3',\n",
    "        'NR2F2',\n",
    "        'NR2F1',\n",
    "        'PEG3',\n",
    "        'SYT1',\n",
    "        'IGFBP5'\n",
    "    )%>%rev()\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d4f7577-726b-41f0-8cec-b70bfc0b6ad9",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Data load"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "c403d829-054a-4207-ae46-f94f46cce701",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading objects:\n",
      "  cell.type.colors\n",
      "  cell.types\n",
      "  cell.types.nohighmito\n",
      "  chr_df\n",
      "  donor_sex\n",
      "  fig\n",
      "  flora_paper_list\n",
      "  gene_type.df\n",
      "  imprinted.df\n",
      "  imprinted.genes\n",
      "  monkey_paper_list\n",
      "  monkey.genes\n",
      "  monkey.markers\n",
      "  nescreg.genes\n",
      "  nescreg.genes.no_trg\n",
      "  nescreg.genes.no_trg.early\n",
      "  nescreg.list\n",
      "  nescreg.list.no_trg\n",
      "  nescreg.list.no_trg.early\n",
      "  nescreg.markers\n",
      "  nescreg.markers.no_trg\n",
      "  nescreg.markers.no_trg.early\n",
      "  nicola_marker_list\n",
      "  origin_sex.colors\n",
      "  pat.de\n",
      "  pat.de.top20\n",
      "  pat.list\n",
      "  pat.neu.de\n",
      "  pat.neu.de.top20\n",
      "  pat.neu.list\n",
      "  pat.neu.top20\n",
      "  pat.top20\n",
      "  phases\n",
      "  phases.colors\n",
      "  protein_coding.genes\n",
      "  region.genes\n",
      "  region.list\n",
      "  region.markers\n",
      "  samples\n",
      "  samples.colors\n",
      "  sex_chr.genes\n",
      "  sex_chr.genes.x\n",
      "  sex_chr.genes.y\n",
      "  shown_2b_list\n",
      "  shown_4d_list\n",
      "  tf.genes\n",
      "  tfs.df\n",
      "  top100.sub.pat\n",
      "  top100.subtype\n",
      "  valid_chr\n",
      "Loading objects:\n",
      "  exp.filt.opts\n",
      "  phase.filt.opts\n",
      "  pval.filt.opts\n",
      "  tf.filt.opts\n",
      "  filt.grid\n",
      "  filt.grid.desc\n"
     ]
    }
   ],
   "source": [
    "load('../data/commons.rda', verbose = T)\n",
    "load('../data/filter.list.rda', verbose=T)\n",
    "\n",
    "subset.top <- readRDS('../data/subset.top.fun.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "af2000ff-17ef-4a59-b17a-161fbcec342f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(\"../plots\"):\n",
      "“'../plots' already exists”\n"
     ]
    }
   ],
   "source": [
    "today_dir <- glue::glue('../plots/dotplots_{Sys.Date()}/')\n",
    "\n",
    "dir.create('../plots')\n",
    "dir.create(today_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "2885d80b-b9b7-4c60-b56a-dcb37c05d921",
   "metadata": {},
   "outputs": [],
   "source": [
    "original.dir <- paste0(today_dir, 'not_scaled/')\n",
    "dir.create(original.dir)\n",
    "\n",
    "clipped.dir <- paste0(today_dir, 'clipped_pct/')\n",
    "dir.create(clipped.dir)\n",
    "\n",
    "scaled.dir <- paste0(today_dir, 'scaled_pct/')\n",
    "dir.create(scaled.dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c6d0a69-50e2-4b33-9eda-8078a1f023cc",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Expression data: no high-mito ~& no 384 donor~"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "454fb401-ce02-4510-bc5e-5bb608ec8515",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Read\n",
    "seu <- readRDS('../data/DE_DS.NoHighMito.pp.scaled.RDS')"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d77600e2-fa50-47b1-9093-21ae624a1c4e",
   "metadata": {},
   "source": [
    "seu.no384 <- readRDS('../data/DE_DS.no_ASD_384.pp.scaled.RDS')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e2500cb9-c3d0-4b75-a140-be10f9d18c10",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] Mes prog\n",
      "7 Levels: FGF17-like Mes prog RG early RG late Neral crest/Mes ... High-mito\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Centering and scaling data matrix\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] RG early\n",
      "7 Levels: FGF17-like Mes prog RG early RG late Neral crest/Mes ... High-mito\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Centering and scaling data matrix\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] Neuron\n",
      "7 Levels: FGF17-like Mes prog RG early RG late Neral crest/Mes ... High-mito\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Centering and scaling data matrix\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] RG late\n",
      "7 Levels: FGF17-like Mes prog RG early RG late Neral crest/Mes ... High-mito\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Centering and scaling data matrix\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] FGF17-like\n",
      "7 Levels: FGF17-like Mes prog RG early RG late Neral crest/Mes ... High-mito\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Centering and scaling data matrix\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] Neral crest/Mes\n",
      "7 Levels: FGF17-like Mes prog RG early RG late Neral crest/Mes ... High-mito\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Centering and scaling data matrix\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ct_seus <- lapply(unique(seu$anno_cluster_fct), function(x){\n",
    "    print(x)\n",
    "    ScaleData(subset(seu, anno_cluster_fct==x), verbose=T)\n",
    "})\n",
    "\n",
    "names(ct_seus) <- unique(seu$anno_cluster_fct)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc3e4f27-59c4-4e24-85e0-5cbc80264817",
   "metadata": {},
   "source": [
    "## Markers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f81f69cd-56ed-481f-8203-72f6eb2fc415",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Seurat.All'</li><li>'Seurat.All.TFs'</li><li>'DESeq2.RGe'</li><li>'DESeq2.RGe.TFs'</li><li>'DESeq2.RGe.PerPhase'</li><li>'DESeq2.RGe.PerPhase.TFs'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Seurat.All'\n",
       "\\item 'Seurat.All.TFs'\n",
       "\\item 'DESeq2.RGe'\n",
       "\\item 'DESeq2.RGe.TFs'\n",
       "\\item 'DESeq2.RGe.PerPhase'\n",
       "\\item 'DESeq2.RGe.PerPhase.TFs'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Seurat.All'\n",
       "2. 'Seurat.All.TFs'\n",
       "3. 'DESeq2.RGe'\n",
       "4. 'DESeq2.RGe.TFs'\n",
       "5. 'DESeq2.RGe.PerPhase'\n",
       "6. 'DESeq2.RGe.PerPhase.TFs'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Seurat.All\"              \"Seurat.All.TFs\"         \n",
       "[3] \"DESeq2.RGe\"              \"DESeq2.RGe.TFs\"         \n",
       "[5] \"DESeq2.RGe.PerPhase\"     \"DESeq2.RGe.PerPhase.TFs\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "openxlsx::getSheetNames('../results/selected_markers/ASD_vs_Ctrl_markers.Annotated.Filter_None.xlsx')\n",
    "\n",
    "markers.all <- openxlsx::read.xlsx('../results/selected_markers/ASD_vs_Ctrl_markers.Annotated.Filter_None.xlsx',\n",
    "                                   sheet='Seurat.All')\n",
    "markers.rge <- openxlsx::read.xlsx('../results/selected_markers/ASD_vs_Ctrl_markers.Annotated.Filter_None.xlsx',\n",
    "                                   sheet='DESeq2.RGe')\n",
    "markers.phase <- openxlsx::read.xlsx('../results/selected_markers/ASD_vs_Ctrl_markers.Annotated.Filter_None.xlsx',\n",
    "                                     sheet='DESeq2.RGe.PerPhase')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "842b1a6f-aba3-495c-8d96-d6d504a54796",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'filter.2of3.over.2of3'</li><li>'filter.2of3.over.max'</li><li>'filter.all.over.2of3'</li><li>'filter.all.over.max'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'filter.2of3.over.2of3'\n",
       "\\item 'filter.2of3.over.max'\n",
       "\\item 'filter.all.over.2of3'\n",
       "\\item 'filter.all.over.max'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'filter.2of3.over.2of3'\n",
       "2. 'filter.2of3.over.max'\n",
       "3. 'filter.all.over.2of3'\n",
       "4. 'filter.all.over.max'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"filter.2of3.over.2of3\" \"filter.2of3.over.max\"  \"filter.all.over.2of3\" \n",
       "[4] \"filter.all.over.max\"  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(markers.all) %>% grep(pattern='filter', value=T) -> av.filters\n",
    "av.filters"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e62865d-7520-4341-bede-4d5b7a9ba734",
   "metadata": {},
   "source": [
    "## Genes to plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "c88b1bb7-f43d-40f6-9ef6-0657dd937a42",
   "metadata": {},
   "outputs": [],
   "source": [
    "filter.de.results <- function(df,\n",
    "                              pval.col, \n",
    "                              logfc.col,\n",
    "                              use.abs.logfc=FALSE,\n",
    "                              \n",
    "                              max.pval=0.05, \n",
    "                              min.logfc=log2(1),\n",
    "                              \n",
    "                              top.by='pval', \n",
    "                              top.n=50, \n",
    "                              ties.method='first',\n",
    "                              add.pars = T,\n",
    "                              split.by=NULL){\n",
    "    \n",
    "    # split analysis\n",
    "    if (!is.null(split.by)){\n",
    "        \n",
    "        split(df, df[,split.by]) %>%\n",
    "        lapply(FUN = function(d){\n",
    "            filter.de.results(df=d,\n",
    "                              pval.col=pval.col, \n",
    "                              logfc.col=logfc.col,\n",
    "                              use.abs.logfc=use.abs.logfc,\n",
    "                              max.pval=max.pval,\n",
    "                              min.logfc=min.logfc,\n",
    "                              top.by=top.by,\n",
    "                              top.n=top.n,\n",
    "                              ties.method=ties.method,\n",
    "                              add.pars=add.pars)\n",
    "        }) %>% do.call(what='rbind') %>%\n",
    "        as.data.frame %>% return\n",
    "        \n",
    "    } else {\n",
    "        \n",
    "        # 1. set columns \n",
    "        df['pval.use'] <- df[,pval.col]\n",
    "        df['logfc.use'] <- df[,logfc.col]\n",
    "        # 2. Only positive markers?\n",
    "        if (use.abs.logfc) {df['logfc.used'] <- abs(df['logfc.used'])}\n",
    "        # 3. add filters \n",
    "        df <- df %>% mutate(\n",
    "            pval.isnumeric = is.numeric(pval.use)&!is.na(pval.use),\n",
    "            logfc.isnumeric = is.numeric(logfc.use)&!is.na(logfc.use),\n",
    "            # 4. thresholds\n",
    "            pval.underormax = pval.use <= max.pval,\n",
    "            logfc.overormin = logfc.use >= min.logfc,\n",
    "            # 5. obtain valid values to rank\n",
    "            rank.consider = pval.isnumeric&logfc.isnumeric&pval.underormax&logfc.overormin,\n",
    "            pval.only_valid = ifelse(rank.consider, pval.use, NA),\n",
    "            logfc.only_valid = ifelse(rank.consider, logfc.use, NA),\n",
    "            # 6. rank values\n",
    "            pval.rank = rank(pval.only_valid, na.last = T, ties.method = ties.method),\n",
    "            logfc.rank = rank(-logfc.only_valid, na.last = T, ties.method = ties.method)\n",
    "        )\n",
    "        # get top\n",
    "        df$pval.top <- df$pval.rank <= top.n\n",
    "        df$logfc.top <- df$logfc.rank <= top.n\n",
    "        # combine filters\n",
    "        df$all.ok <- df$rank.consider & df[,paste0(top.by, '.top')]\n",
    "        \n",
    "        if (add.pars){\n",
    "            df %>% mutate(\n",
    "                pval.col=pval.col, \n",
    "                logfc.col=logfc.col,\n",
    "                use.abs.logfc=use.abs.logfc,\n",
    "                max.pval=max.pval,\n",
    "                min.logfc=min.logfc,\n",
    "                top.by=top.by,\n",
    "                top.n=top.n,\n",
    "                ties.method=ties.method)\n",
    "        }\n",
    "        \n",
    "        return(df)\n",
    "        \n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a12986ba-3124-401d-96fa-375ccd9391e5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "a8affef0-86b8-4165-a7f9-1e90984c6e4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "figure_selection.a <- c(\n",
    "    list('early AV'=c(nicola_marker_list[[1]][1:5], \n",
    "                      rev(nicola_marker_list[[1]])[2])),\n",
    "    monkey.markers[-(1:3)]) %>% lapply(sort)\n",
    "\n",
    "figure_selection.b <- list(\n",
    "    'FR' = c('SP8', 'WNT7B', 'ZIC5', 'ETV5'),\n",
    "    'MS' = c('HEY1', 'SOX3', 'EMX2', 'LAMTOR3', 'CREB5'),\n",
    "    'OC' = c('FGF13', 'IGF2', 'ZBTB16','FEZF1', 'HEY1'),\n",
    "    'Tem' = c('PEG3', 'NR2F1', 'FGFR3', 'NR2F2', 'RSPO3') ) %>% lapply(sort)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "7fcdb6a6-f8a7-42c0-8a19-2b991a90ae5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# define lists of genes to plot\n",
    "\n",
    "sex_chr.list <- with(subset(chr_df, chr %in% c('X', 'Y')),{split(hgnc_symbol, chr)})\n",
    "\n",
    "imprinted.list <- split(imprinted.df$gene, ifelse(imprinted.df$Expressed.Allele=='','Unknown',imprinted.df$Expressed.Allele))\n",
    "imprinted.hq.list <- with(subset(imprinted.df, Imprinted.Status == 'Imprinted'),{split(gene, ifelse(Expressed.Allele=='','Unknown',Expressed.Allele))})\n",
    "\n",
    "\n",
    "\n",
    "genelists.to.plot <- list(\n",
    "    \n",
    "    \n",
    "#     'Selected by Nicola M' = nicola_marker_list,\n",
    "#     'JAN 31 Selected by Nicola M' = nicola.selection2,\n",
    "    # 'FEB 09 Reduced by Nicola M ' = nicola.selection3\n",
    "    \n",
    "#     'Figure selection 1' = figure_selection.a,\n",
    "#     'Figure selection 2' = figure_selection.b,\n",
    "    \n",
    "    \n",
    "#     'Sex chromosome genes' = sex_chr.list,\n",
    "#     'Imprinted genes' = imprinted.hq.list,\n",
    "#     'Imprinted genes - with predictions' = imprinted.list,\n",
    "    \n",
    "#     'Flora paper'=flora_paper_list,\n",
    "#     'Monkey genes in text' = monkey_paper_list,\n",
    "#     'Flora and RPC, AV, GE (Monkey - Fig2B)' = monkey.markers,\n",
    "#     'Flora' = monkey.markers[-(1:3)],\n",
    "    \n",
    "    'RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)' = monkey.markers[1:3]\n",
    "    \n",
    "#     'Markers of patterning centers - with Rest (Monkey - TableS2.subtype_markers)' = top100.sub.pat,\n",
    "#     'Shown Markers of patterning centers (Monkey - Fig2B shown)' = shown_2b_list,\n",
    "    # 'Markers of patterning centers (Monkey - TableS3.organizer_markers)' = pat.list,\n",
    "#     'Top 20 Markers of patterning centers (Monkey - TableS3.organizer_markers)' = pat.top20,\n",
    "#     'Markers of patterning centers - with NESC (Monkey - TableS4.organizer_markers_versus_NSCs)' = pat.neu.list,\n",
    "#     'Top 20 Markers of patterning centers - with NESC (Monkey - TableS4.organizer_markers_versus_NSCs)' = pat.neu.top20,\n",
    "    \n",
    "    \n",
    "    # 'NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)' = nicola.selection4\n",
    "    # 'NESC progression region markers (Monkey - Fig4D shown)' = shown_4d_list,\n",
    "    # 'NESC progression region markers (Monkey - Fig4D.TableS6.Region_markers_in_vent_NSC_progr)' = nescreg.list,\n",
    "    'NESC progression region markers - no tRG (Monkey - Fig4D.TableS6.Region_markers_in_vent_NSC_progr)' = nescreg.list.no_trg,\n",
    "    # 'NESC progression region markers - no tRG - early (Monkey - Fig4D.TableS6.Region_markers_in_vent_NSC_progr)' = nescreg.list.no_trg.early\n",
    "    # 'PC markers' = filt1.list\n",
    "#     'Regional markers (Monkey - TableS7.regional_enrichment_patterns_in_multiple_regions)' = region.list\n",
    ") %>% lapply(function(x){lapply(x, rev)})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "17283f32-f3da-4734-a212-deb74b605243",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# set ordering in all lists to follow maximal sample of expression in early radial glia\n",
    "genelists.to.plot %>% lapply(FUN=function(x){\n",
    "    unique(unlist(x)) -> g\n",
    "    d <- DotPlot(subset(seu, anno_cluster_fct == 'RG early'), \n",
    "                 features = g,\n",
    "                 group.by='cell_origin')\n",
    "    # first group to obtain max sample per gene\n",
    "    d$data %>% group_by(features.plot) %>% summarise(ms=id[which.max(pct.exp)], exp=max(pct.exp)) %>%\n",
    "    # then group by maxsample\n",
    "    group_by(ms) %>% arrange(ms, desc(exp)) %>% ungroup()%>% select(features.plot) %>%\n",
    "    apply(FUN=as.character, 1) %>% rev() -> glevels\n",
    "\n",
    "}) -> genelevels\n",
    "\n",
    "for (ln in names(genelevels)){\n",
    "    ls <- genelevels[[ln]]\n",
    "    genelists.to.plot[[ln]] <- lapply(genelists.to.plot[[ln]], function(x){\n",
    "        x[order(match(x,ls))]\n",
    "    })\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "b1ce4352-2e60-43a0-8551-316ae1103411",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`</strong> = <dl>\n",
       "\t<dt>$RPC</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'FGF18'</li><li>'SP8'</li><li>'ZIC3'</li><li>'FGF3'</li><li>'FGF17'</li><li>'ZIC4'</li><li>'ZIC1'</li><li>'FGF8'</li><li>'SPRY2'</li><li>'ETV1'</li><li>'DLK1'</li></ol>\n",
       "</dd>\n",
       "\t<dt>$AV</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'OLIG2'</li><li>'SHH'</li><li>'NKX2-1'</li><li>'DLX2'</li><li>'SOX6'</li><li>'SOX3'</li><li>'OLIG1'</li><li>'ASCL1'</li><li>'NKX6-2'</li><li>'FOXG1'</li><li>'DLK1'</li></ol>\n",
       "</dd>\n",
       "\t<dt>$GE</dt>\n",
       "\t\t<dd><style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'OLIG2'</li><li>'GSX2'</li><li>'ZEB1'</li><li>'PBX3'</li><li>'ZEB2'</li><li>'MAP2'</li><li>'POU3F3'</li><li>'FGFR2'</li><li>'POU3F2'</li><li>'FOXG1'</li><li>'MEIS2'</li></ol>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\textbf{\\$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`} = \\begin{description}\n",
       "\\item[\\$RPC] \\begin{enumerate*}\n",
       "\\item 'FGF18'\n",
       "\\item 'SP8'\n",
       "\\item 'ZIC3'\n",
       "\\item 'FGF3'\n",
       "\\item 'FGF17'\n",
       "\\item 'ZIC4'\n",
       "\\item 'ZIC1'\n",
       "\\item 'FGF8'\n",
       "\\item 'SPRY2'\n",
       "\\item 'ETV1'\n",
       "\\item 'DLK1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$AV] \\begin{enumerate*}\n",
       "\\item 'OLIG2'\n",
       "\\item 'SHH'\n",
       "\\item 'NKX2-1'\n",
       "\\item 'DLX2'\n",
       "\\item 'SOX6'\n",
       "\\item 'SOX3'\n",
       "\\item 'OLIG1'\n",
       "\\item 'ASCL1'\n",
       "\\item 'NKX6-2'\n",
       "\\item 'FOXG1'\n",
       "\\item 'DLK1'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$GE] \\begin{enumerate*}\n",
       "\\item 'OLIG2'\n",
       "\\item 'GSX2'\n",
       "\\item 'ZEB1'\n",
       "\\item 'PBX3'\n",
       "\\item 'ZEB2'\n",
       "\\item 'MAP2'\n",
       "\\item 'POU3F3'\n",
       "\\item 'FGFR2'\n",
       "\\item 'POU3F2'\n",
       "\\item 'FOXG1'\n",
       "\\item 'MEIS2'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "**$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`** = $RPC\n",
       ":   1. 'FGF18'\n",
       "2. 'SP8'\n",
       "3. 'ZIC3'\n",
       "4. 'FGF3'\n",
       "5. 'FGF17'\n",
       "6. 'ZIC4'\n",
       "7. 'ZIC1'\n",
       "8. 'FGF8'\n",
       "9. 'SPRY2'\n",
       "10. 'ETV1'\n",
       "11. 'DLK1'\n",
       "\n",
       "\n",
       "\n",
       "$AV\n",
       ":   1. 'OLIG2'\n",
       "2. 'SHH'\n",
       "3. 'NKX2-1'\n",
       "4. 'DLX2'\n",
       "5. 'SOX6'\n",
       "6. 'SOX3'\n",
       "7. 'OLIG1'\n",
       "8. 'ASCL1'\n",
       "9. 'NKX6-2'\n",
       "10. 'FOXG1'\n",
       "11. 'DLK1'\n",
       "\n",
       "\n",
       "\n",
       "$GE\n",
       ":   1. 'OLIG2'\n",
       "2. 'GSX2'\n",
       "3. 'ZEB1'\n",
       "4. 'PBX3'\n",
       "5. 'ZEB2'\n",
       "6. 'MAP2'\n",
       "7. 'POU3F3'\n",
       "8. 'FGFR2'\n",
       "9. 'POU3F2'\n",
       "10. 'FOXG1'\n",
       "11. 'MEIS2'\n",
       "\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`\n",
       "$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`$RPC\n",
       " [1] \"FGF18\" \"SP8\"   \"ZIC3\"  \"FGF3\"  \"FGF17\" \"ZIC4\"  \"ZIC1\"  \"FGF8\"  \"SPRY2\"\n",
       "[10] \"ETV1\"  \"DLK1\" \n",
       "\n",
       "$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`$AV\n",
       " [1] \"OLIG2\"  \"SHH\"    \"NKX2-1\" \"DLX2\"   \"SOX6\"   \"SOX3\"   \"OLIG1\"  \"ASCL1\" \n",
       " [9] \"NKX6-2\" \"FOXG1\"  \"DLK1\"  \n",
       "\n",
       "$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`$GE\n",
       " [1] \"OLIG2\"  \"GSX2\"   \"ZEB1\"   \"PBX3\"   \"ZEB2\"   \"MAP2\"   \"POU3F3\" \"FGFR2\" \n",
       " [9] \"POU3F2\" \"FOXG1\"  \"MEIS2\" \n",
       "\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<strong>$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`</strong> = <style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'OLIG2'</li><li>'SHH'</li><li>'NKX2-1'</li><li>'GSX2'</li><li>'FGF18'</li><li>'DLX2'</li><li>'SOX6'</li><li>'SP8'</li><li>'ZEB1'</li><li>'PBX3'</li><li>'SOX3'</li><li>'ZEB2'</li><li>'MAP2'</li><li>'OLIG1'</li><li>'ASCL1'</li><li>'NKX6-2'</li><li>'ZIC3'</li><li>'POU3F3'</li><li>'FGFR2'</li><li>'POU3F2'</li><li>'FGF3'</li><li>'FGF17'</li><li>'ZIC4'</li><li>'ZIC1'</li><li>'FGF8'</li><li>'SPRY2'</li><li>'ETV1'</li><li>'FOXG1'</li><li>'MEIS2'</li><li>'DLK1'</li></ol>\n"
      ],
      "text/latex": [
       "\\textbf{\\$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`} = \\begin{enumerate*}\n",
       "\\item 'OLIG2'\n",
       "\\item 'SHH'\n",
       "\\item 'NKX2-1'\n",
       "\\item 'GSX2'\n",
       "\\item 'FGF18'\n",
       "\\item 'DLX2'\n",
       "\\item 'SOX6'\n",
       "\\item 'SP8'\n",
       "\\item 'ZEB1'\n",
       "\\item 'PBX3'\n",
       "\\item 'SOX3'\n",
       "\\item 'ZEB2'\n",
       "\\item 'MAP2'\n",
       "\\item 'OLIG1'\n",
       "\\item 'ASCL1'\n",
       "\\item 'NKX6-2'\n",
       "\\item 'ZIC3'\n",
       "\\item 'POU3F3'\n",
       "\\item 'FGFR2'\n",
       "\\item 'POU3F2'\n",
       "\\item 'FGF3'\n",
       "\\item 'FGF17'\n",
       "\\item 'ZIC4'\n",
       "\\item 'ZIC1'\n",
       "\\item 'FGF8'\n",
       "\\item 'SPRY2'\n",
       "\\item 'ETV1'\n",
       "\\item 'FOXG1'\n",
       "\\item 'MEIS2'\n",
       "\\item 'DLK1'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "**$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`** = 1. 'OLIG2'\n",
       "2. 'SHH'\n",
       "3. 'NKX2-1'\n",
       "4. 'GSX2'\n",
       "5. 'FGF18'\n",
       "6. 'DLX2'\n",
       "7. 'SOX6'\n",
       "8. 'SP8'\n",
       "9. 'ZEB1'\n",
       "10. 'PBX3'\n",
       "11. 'SOX3'\n",
       "12. 'ZEB2'\n",
       "13. 'MAP2'\n",
       "14. 'OLIG1'\n",
       "15. 'ASCL1'\n",
       "16. 'NKX6-2'\n",
       "17. 'ZIC3'\n",
       "18. 'POU3F3'\n",
       "19. 'FGFR2'\n",
       "20. 'POU3F2'\n",
       "21. 'FGF3'\n",
       "22. 'FGF17'\n",
       "23. 'ZIC4'\n",
       "24. 'ZIC1'\n",
       "25. 'FGF8'\n",
       "26. 'SPRY2'\n",
       "27. 'ETV1'\n",
       "28. 'FOXG1'\n",
       "29. 'MEIS2'\n",
       "30. 'DLK1'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$`RPC, AV, GE (Monkey - Fig2B shown RPC,AV,GE)`\n",
       " [1] \"OLIG2\"  \"SHH\"    \"NKX2-1\" \"GSX2\"   \"FGF18\"  \"DLX2\"   \"SOX6\"   \"SP8\"   \n",
       " [9] \"ZEB1\"   \"PBX3\"   \"SOX3\"   \"ZEB2\"   \"MAP2\"   \"OLIG1\"  \"ASCL1\"  \"NKX6-2\"\n",
       "[17] \"ZIC3\"   \"POU3F3\" \"FGFR2\"  \"POU3F2\" \"FGF3\"   \"FGF17\"  \"ZIC4\"   \"ZIC1\"  \n",
       "[25] \"FGF8\"   \"SPRY2\"  \"ETV1\"   \"FOXG1\"  \"MEIS2\"  \"DLK1\"  \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "genelists.to.plot\n",
    "genelevels"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db9315a5-3f65-4ce1-a2d8-54d059f38ea3",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c9667780-0a1c-4292-861a-b59b29d6bb23",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# Dotplot functions"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b00aa345-ebf5-4ac1-9e5d-285211b5c292",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "## Helpers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "777ae638-50ce-4d14-bcec-0c7f87ad1169",
   "metadata": {},
   "outputs": [],
   "source": [
    "scale_zero_to_max <- function(x){\n",
    "    \n",
    "    if (any(x<0)){stop('Does not work with negative numbers')}\n",
    "    \n",
    "    max.x <- max(x, na.rm=T)\n",
    "    \n",
    "    # If everything is zero, return original\n",
    "    if (max.x == 0){\n",
    "        return(x)\n",
    "    } else {\n",
    "        # otherwise\n",
    "        return(x/max.x)\n",
    "    }\n",
    "}\n",
    "scale_min_to_max <- function(x){\n",
    "        \n",
    "    max.x <- max(x, na.rm=T)\n",
    "    if (max.x == 0){\n",
    "        return(x)\n",
    "    } else {\n",
    "        min.x <- min(x)\n",
    "        # otherwise\n",
    "        return((x-min.x)/(max.x-min.x))\n",
    "    }\n",
    "}\n",
    "\n",
    "labels_zero_to_max <- function(x){\n",
    "    # message(x)\n",
    "    y <- scales::label_percent()(x)\n",
    "    # message(y)\n",
    "    y[length(y)] <- 'Max'\n",
    "    return(y)\n",
    "}\n",
    "\n",
    "labels_min_to_max <- function(x){\n",
    "    # messsage(x)\n",
    "    y <- scales::label_percent()(x)\n",
    "    # messsage(y)\n",
    "    y[1] <- 'Min'\n",
    "    y[length(y)] <- 'Max'\n",
    "    return(y)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be990e03-4b3e-486b-b5ea-4501cf0add3b",
   "metadata": {},
   "source": [
    "## Dotplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "id": "3a43e07d-e2c9-4ce1-b1ba-ad3c4585558e",
   "metadata": {},
   "outputs": [],
   "source": [
    "dotplot <- function(seu, gene.lists, \n",
    "                    repeated.do = 'label', force_hide_repeated=T,\n",
    "                    exp.slot = 'scale.data',\n",
    "                    \n",
    "                    min.col='white', max.col='orangered3',\n",
    "                    min.point.size=.1, max.point.size=3, \n",
    "                    \n",
    "                    pct.min=3, n.samples.over.pct=1,\n",
    "                    \n",
    "                    clip.upper.pct=100, n.breaks.pct=6,\n",
    "                    \n",
    "                    pct_scale_by=NULL, pct_scaler = 'zero_to_max',\n",
    "                    exp_scale_by=NULL, exp_scaler = 'min_to_max',\n",
    "                    \n",
    "                    cluster.genes=T, max.clusters=6, cluster_on='pct', \n",
    "                    \n",
    "                    title=NULL, subtitle=NULL, y.lab='genes', \n",
    "                    ...){\n",
    "\n",
    "    require(ggh4x)\n",
    "    facet.vars <- vars(feature.groups)\n",
    "    \n",
    "    exp.is_scaled <- F\n",
    "    pct.is_scaled <- F\n",
    "    pct_labeller <- function(...){return(as.character(...))}\n",
    "    exp_labeller <- function(...){return(as.character(...))}\n",
    "    \n",
    "    \n",
    "    ## PROCESS WITH REPEATED GENES\n",
    "    repeated.genes <- list()\n",
    "    nr.genes <- list()\n",
    "    for (l in names(gene.lists)){\n",
    "        clist <- gene.lists[names(gene.lists)!=l]\n",
    "        repeated <- gene.lists[[l]]%in%unique(unlist(clist))\n",
    "        print(glue::glue('{sum(repeated)} ({round(sum(repeated)/length(gene.lists[[l]]), 4)*100}%) repeated genes in {l} list'))\n",
    "        nr.genes[[l]] <- gene.lists[[l]][!repeated]\n",
    "        repeated.genes[[l]] <- gene.lists[[l]][repeated]\n",
    "    }\n",
    "    # remove empty lists\n",
    "    nr.genes <- nr.genes[sapply(nr.genes, length)!=0]\n",
    "    \n",
    "    # REMOVE OR ADD\n",
    "    if ((repeated.do == 'label') & (sum(sapply(repeated.genes, length))>0)){\n",
    "        nr.genes$repeated <- unique(unlist(repeated.genes))\n",
    "        facet.vars <- vars(feature.groups, repeated)\n",
    "    } else if (!repeated.do%in%c('remove', 'label')){\n",
    "        message(glue::glue('\"{repeated.do}\" is not a valid `repeated.do` option [\"remove\", \"label\"]. Removing...'))\n",
    "    }\n",
    "    \n",
    "    \n",
    "    # ANY VALID GENES ?\n",
    "    nplot.genes <- sum(sapply(nr.genes, length)) \n",
    "    if (nplot.genes==0){return(NULL)}\n",
    "    \n",
    "    # FETCH DATA FROM DOTPLOT\n",
    "    p <- DotPlot(seu, \n",
    "                 group.by='anno_cluster_fct', split.by='cell_origin', \n",
    "                 features=nr.genes, cols = 'Reds', ...)\n",
    "    \n",
    "    pdata <- p$data$id%>% as.character%>% strsplit(split='_D8_', fixed=T) %>%\n",
    "        as.data.frame() %>% t()%>% as.data.frame() %>%\n",
    "        mutate(celltype=factor(V1, cell.types.nohighmito), \n",
    "               Donor=factor(paste0('D8_', V2), samples)) %>%\n",
    "        cbind(p$data, .) %>% as.data.frame()\n",
    "    \n",
    "    \n",
    "    # REMOVE LOW EXP BASED ON PCT\n",
    "    if ( is.numeric(pct.min)&is.numeric(n.samples.over.pct)&(n.samples.over.pct>0)&(pct.min>0) ){\n",
    "        message(glue::glue('Requiring {pct.min}% expression in at least {n.samples.over.pct} group(s)'))\n",
    "        \n",
    "        (pdata%>% \n",
    "         mutate(pct.over=pct.exp>pct.min)%>% group_by(features.plot) %>% summarise(n.pct.over = sum(pct.over)) %>%  subset(n.pct.over>=n.samples.over.pct)\n",
    "        )$features.plot -> exp.genes\n",
    "        \n",
    "        \n",
    "        print(glue::glue('Number of genes {length(unique(pdata$features.plot))} -> {length(exp.genes)}'))\n",
    "        pdata <- subset(pdata, features.plot %in% exp.genes)\n",
    "        \n",
    "    } \n",
    "    # ADD CORRESPONDING REPEATED GENES\n",
    "    if ((repeated.do == 'label')&(sum(sapply(repeated.genes, length))>0)){\n",
    "        p.repeated <- subset(pdata, feature.groups == 'repeated')\n",
    "        pdata <- subset(pdata, feature.groups != 'repeated') \n",
    "\n",
    "        lapply(names(repeated.genes), function(x){\n",
    "            subset(p.repeated, features.plot %in% repeated.genes[[x]])%>%mutate(feature.groups=factor(x, names(gene.lists)), repeated='Not exclusive')\n",
    "        }) %>% do.call(what='rbind') %>% rbind(pdata%>%mutate(repeated='Exclusive'))%>%\n",
    "        mutate(repeated=factor(repeated, c('Exclusive', 'Not exclusive'))) -> pdata\n",
    "\n",
    "    } else {\n",
    "        pdata$repeated <- factor('Exclusive')\n",
    "    }\n",
    "    \n",
    "    \n",
    "    \n",
    "    \n",
    "    # N GENES UNIQUE\n",
    "    nunique.genes <- length(unique(pdata$features.plot))\n",
    "    \n",
    "    # PROCESS EXPRESSION\n",
    "    # Define expression slot\n",
    "    if(exp.slot=='data'){\n",
    "        pdata$exp <- pdata$avg.exp\n",
    "    } else if (exp.slot == 'scale.data'){\n",
    "        pdata$exp <- pdata$avg.exp.scaled\n",
    "        exp.is_scaled <- T\n",
    "    }\n",
    "    # CLIP PCT\n",
    "    pdata[pdata$pct.exp>clip.upper.pct,'pct.exp'] <- clip.upper.pct \n",
    "\n",
    "    # scales\n",
    "    if (!is.null(pct_scale_by)){\n",
    "        pct_labeller <- if(pct_scaler=='zero_to_max'){labels_zero_to_max}else{labels_min_to_max}\n",
    "        pct_scaler <- if(pct_scaler=='zero_to_max'){scale_zero_to_max}else{scale_min_to_max}\n",
    "        \n",
    "        if (clip.upper.pct!=100){warning('Clipping and scaling pct is not advised')}\n",
    "        pdata <- pdata%>%group_by_at(pct_scale_by) %>% mutate(pct.exp=pct_scaler(pct.exp)) \n",
    "        clip.upper.pct <- 1\n",
    "    }\n",
    "    if (!is.null(exp_scale_by)){\n",
    "        exp.is_scaled <- T\n",
    "        exp_labeller <- if(exp_scaler=='zero_to_max'){labels_zero_to_max}else{labels_min_to_max}\n",
    "        exp_scaler <- if(exp_scaler=='zero_to_max'){scale_zero_to_max}else{scale_min_to_max}\n",
    "        pdata <- pdata%>%group_by_at(pct_scale_by) %>% mutate(exp=exp_scaler(exp)) \n",
    "    }\n",
    "    \n",
    "    \n",
    "    # CLUSTER GENES\n",
    "    if (cluster.genes){\n",
    "        if(nunique.genes>30){\n",
    "            value.var <- if(cluster_on=='pct'){'pct.exp'}else{'exp'}\n",
    "            exp.matrix <- reshape2::dcast(pdata, value.var = value.var,\n",
    "                formula='feature.groups+repeated+features.plot~celltype+Donor')\n",
    "            exp.matrix[is.na(exp.matrix)] <- min(exp.matrix[,-(1:3)], na.rm = T)\n",
    "\n",
    "            require(amap)\n",
    "            for (i in seq(from=min(max.clusters, nrow(exp.matrix)), to=1)){\n",
    "                kms <- tryCatch({amap::Kmeans(x = exp.matrix[,-(1:3)], centers = i,iter.max = 100)},\n",
    "                               error=function(...){NULL})\n",
    "                if (!is.null(kms)){\n",
    "                    pdata <- merge(pdata, cbind(exp.matrix[,1:3], cluster=kms$cluster), \n",
    "                                   by=c('feature.groups', 'repeated', 'features.plot'), all=T)\n",
    "                    facet.vars <- if (repeated.do=='label'){\n",
    "                        vars(cluster, feature.groups, repeated)}else{\n",
    "                        vars(feature.groups, cluster)}    \n",
    "                    break()\n",
    "                }\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "    \n",
    "\n",
    "    \n",
    "    # REMOVE REPEATED GROUP IF THERE IS NONE\n",
    "    rep.fac <- grepl(pattern='repeated', \n",
    "                     gsub(pattern='~', replacement='', fixed=T, as.character(facet.vars)))\n",
    "    if (any(rep.fac) & (force_hide_repeated | all(pdata$repeated=='Exclusive'))){\n",
    "        facet.vars <- facet.vars[!rep.fac]\n",
    "    }\n",
    "\n",
    "    \n",
    "    \n",
    "    # PLOT\n",
    "        # MAPPING, AXES, FACETS\n",
    "    p <- ggplot(pdata, \n",
    "                aes(x=Donor, y=features.plot,\n",
    "                    size = pct.exp, fill = exp)) +\n",
    "        scale_x_discrete(position = 'top') + \n",
    "        facet_nested(rows=facet.vars, cols=vars(celltype),\n",
    "                     strip = strip_nested(size = 'variable'), \n",
    "                     scales='free', space='free', switch='y') + \n",
    "        # DOTS\n",
    "        geom_point(color='#222222FF', stroke=.05, shape=21) +\n",
    "        # SCALES\n",
    "        scale_fill_gradient(low = min.col, high=max.col, labels=exp_labeller)+\n",
    "        scale_size(range=c(min.point.size, max.point.size),\n",
    "                   labels=pct_labeller,\n",
    "                   limits=c(0,clip.upper.pct), \n",
    "                   breaks=seq(from=0, to=clip.upper.pct, length.out = n.breaks.pct)) +\n",
    "        # LEGEND\n",
    "        guides(size=guide_legend(override.aes = list(fill=max.col), \n",
    "                                 title.position = 'top', nrow = 2),\n",
    "               fill=guide_colorbar(title.position = 'top',\n",
    "                                   direction = 'horizontal')) +\n",
    "        # NAMES\n",
    "        labs(y=y.lab, x='Celltype and donor',\n",
    "             size=paste0(if(!is.null(pct_scale_by)){'Scaled '}else{''}, \"Percent Expressed\"), \n",
    "             fill=paste0(if(exp.is_scaled){'Scaled '}else{''}, \"Average Expression\")) + \n",
    "        # THEMING\n",
    "        theme_linedraw() + \n",
    "        theme(panel.grid.major = element_line(color='#66666644', linewidth=0.2),\n",
    "              strip.placement='outside',\n",
    "              strip.text.x = element_text(color='black'),\n",
    "              strip.text.y.left = element_text(color='black', angle=0),\n",
    "              strip.background = element_rect(fill='#88888822'),\n",
    "              axis.text.x.top=element_text(angle=90, hjust=0, vjust=0.5),\n",
    "              legend.position='top')\n",
    "    \n",
    "    if(!is.null(title) | !is.null(subtitle)){\n",
    "        p <- p + ggtitle(title, subtitle=subtitle)\n",
    "    }\n",
    "    \n",
    "    return(p)\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "0ad3c19d-8db8-409e-8dce-b782fe6b1f1e",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# p <- dotplot(seu = seu, n.breaks.pct = 6,\n",
    "#              # pct_scale_by = pct_scale_by,\n",
    "#              exp_scale_by = NULL,\n",
    "#              # exp.slot = 'scale.data',\n",
    "#              gene.list = genelists.to.plot[[4]], \n",
    "#              y.lab = names(genelists.to.plot)[4],\n",
    "#              cluster.genes = T,\n",
    "#              # title=NULL, subtitle=NULL,\n",
    "#              # title='Example', subtitle=NULL,\n",
    "#              # title=NULL, subtitle='Example',\n",
    "#              # title='Example', subtitle='Example',\n",
    "#              pct.min = , n.samples.over.pct = 1, \n",
    "#              repeated.do = 'remove'\n",
    "#             )\n",
    "\n",
    "# n.genes.plot <- nrow(unique(p$data[,c('features.plot', 'feature.groups'), drop=F])) + p$data[,names(p$facet$params$rows),drop=F]%>%unique()%>% nrow() * 0.5\n",
    "# row.chars <- p$data[,names(p$facet$params$rows), drop=F]%>%apply(1, function(x){sum(nchar(as.character(x)))}) %>% max\n",
    "# # row.chars <- if (length(names(p$facet$params$rows))==1){max(row.chars)}else{ apply(row.chars, 2, sum)}\n",
    "# print(n.genes.plot)\n",
    "# print(row.chars)\n",
    "\n",
    "# fig(w = width.lm$coefficients['x']*row.chars+width.lm$coefficients[1], \n",
    "#     h= height.lm$coefficients['x']*n.genes.plot+height.lm$coefficients[1])\n",
    "# p "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b4666ff3-fb09-47a3-b610-78a3f9f1f8a0",
   "metadata": {},
   "source": [
    "## Save dotplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "195b2fb1-b61f-427a-b11b-34413da8e7f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# For genes x samples\n",
    "width.lm <- lm(formula = y~x, data = data.frame(\n",
    "    x=c(27, 17),\n",
    "    y=c(10, 9)))\n",
    "# width.lm$coefficients[2]*27 + width.lm$coefficients[1]\n",
    "\n",
    "\n",
    "\n",
    "height.lm <- lm(formula = y~x, data = data.frame(\n",
    "    x=c(1, 228), # ns.genes\n",
    "    y=c(3.3, 30) # output sizes\n",
    "))\n",
    "# height.lm$coefficients[2]*228 + height.lm$coefficients[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 467,
   "id": "dc7a71d3-be02-4ad1-9609-5d1bf9555e8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "dotplot.save <- function(\n",
    "    seu, gene.list, name, save.dir='./', min.width = 0,\n",
    "    with_clustering=T, cluster_on='pct', without_clustering=T,\n",
    "    pct.min = 3, n.samples.over.pct = 1, repeated.do = 'label', genelevels=NULL, ...){\n",
    "    \n",
    "    if (any(sapply(gene.list, function(x){length(x)!=length(unique(x))}))){\n",
    "        stop('check')\n",
    "    }\n",
    "    if (!any(with_clustering, without_clustering)){return()}\n",
    "    \n",
    "    \n",
    "    \n",
    "    dir.create(save.dir, recursive=T)\n",
    "    message(name)\n",
    "    \n",
    "    ps <- list()\n",
    "    \n",
    "    # with clustering\n",
    "    if (with_clustering){\n",
    "        message('--> With clustering...')\n",
    "        p <- dotplot(seu = seu, \n",
    "                     gene.list = gene.list, \n",
    "                     y.lab = name,\n",
    "                     cluster_on=cluster_on,\n",
    "                     cluster.genes = T,\n",
    "                     pct.min = pct.min,\n",
    "                     n.samples.over.pct = n.samples.over.pct, \n",
    "                     repeated.do = repeated.do,\n",
    "                     ...)\n",
    "        \n",
    "        # override order of genes\n",
    "        if (genelevels){\n",
    "            glevels <- genelevels[[l]]\n",
    "            glevels <- glevels[glevels%in%p$data$features.plot]\n",
    "            p$data %>% mutate(features.plot = factor(as.character(features.plot), glevels)) -> p$data\n",
    "        }\n",
    "        \n",
    "        if (!is.null(p)){\n",
    "            n.groups <- p$data[,names(p$facet$params$rows), drop=F]%>%unique()\n",
    "            n.groups <- if(!any(is.null(dim(n.groups)))){nrow(n.groups)}else{length(n.groups)}\n",
    "            n.genes.plot <- nrow(unique(p$data[,c('features.plot', 'feature.groups')])) + n.groups*0.5\n",
    "\n",
    "            n_cts <- length(unique(seu$anno_cluster_fct))\n",
    "            \n",
    "            if(n.genes.plot>=1){\n",
    "\n",
    "                row.chars <- p$data[,names(p$facet$params$rows), drop=F]%>%\n",
    "                    apply(1, function(x){sum(nchar(as.character(x)))}) %>% max\n",
    "                if (row.chars==1){row.chars <- 10}\n",
    "\n",
    "                pdim <- c(\n",
    "                    # width = width.lm$coefficients['x']*row.chars+width.lm$coefficients[1], \n",
    "                    width = width.lm$coefficients['x']*row.chars+width.lm$coefficients[1]*n_cts/6+.5, \n",
    "                    height = height.lm$coefficients['x']*n.genes.plot+height.lm$coefficients[1])\n",
    "                if (pdim[1] < min.width){pdim[1] <- min.width}\n",
    "\n",
    "                fig(pdim['width.x'], pdim['height.x'])\n",
    "                pdf(file = glue::glue('{save.dir}{name}.clustered.pdf'), width=pdim['width.x'], height=pdim['height.x'])\n",
    "                plot(p)\n",
    "                dev.off()\n",
    "                ps[['clustered']] <- p\n",
    "                \n",
    "                p$data%>% select(feature.groups,features.plot) %>% arrange(feature.groups,desc(features.plot)) %>% unique() -> sorted_genes_last \n",
    "                write.csv(sorted_genes_last, glue::glue('{save.dir}{name}.clustered.genelist.csv'),row.names=F)\n",
    "\n",
    "            }\n",
    "        }\n",
    "    }\n",
    "\n",
    "    \n",
    "    # without clustering\n",
    "    if (without_clustering){\n",
    "        message('--> Without clustering...')\n",
    "        p <- dotplot(seu = seu, \n",
    "                     gene.list = gene.list, \n",
    "                     y.lab = name,\n",
    "                     cluster.genes = F,\n",
    "                     pct.min = pct.min,\n",
    "                     n.samples.over.pct = n.samples.over.pct, \n",
    "                     repeated.do = repeated.do,\n",
    "                     ...)\n",
    "        \n",
    "        # override order of genes\n",
    "        glevels <- genelevels[[l]]\n",
    "        glevels <- glevels[glevels%in%p$data$features.plot]\n",
    "        p$data %>% mutate(features.plot = factor(as.character(features.plot), glevels)) -> p$data\n",
    "        \n",
    "        if (!is.null(p)){\n",
    "            n.groups <- p$data[,names(p$facet$params$rows), drop=F]%>%unique()\n",
    "            n.groups <- if(!any(is.null(dim(n.groups)))){nrow(n.groups)}else{length(n.groups)}\n",
    "            n.genes.plot <- nrow(unique(p$data[,c('features.plot', 'feature.groups')])) + n.groups*0.7\n",
    "\n",
    "            n_cts <- length(unique(seu$anno_cluster_fct))\n",
    "            \n",
    "            if(n.genes.plot>=1){\n",
    "\n",
    "                row.chars <- p$data[,names(p$facet$params$rows), drop=F]%>%\n",
    "                    apply(1, function(x){sum(nchar(as.character(x)))}) %>% max\n",
    "                if (row.chars==1){row.chars <- 10}\n",
    "\n",
    "                pdim <- c(\n",
    "                    # width = width.lm$coefficients['x']*row.chars+width.lm$coefficients[1], \n",
    "                    width = width.lm$coefficients['x']*row.chars+width.lm$coefficients[1]*n_cts/6+.5, \n",
    "                    height = height.lm$coefficients['x']*n.genes.plot+height.lm$coefficients[1])\n",
    "                if (pdim[1] < min.width){pdim[1] <- min.width}\n",
    "                fig(pdim['width.x'], pdim['height.x'])\n",
    "\n",
    "                pdf(file = glue::glue('{save.dir}{name}.pdf'), width=pdim['width.x'], height=pdim['height.x'])    \n",
    "                plot(p)\n",
    "                dev.off()\n",
    "                ps[['not clust']] <- p\n",
    "                \n",
    "                p$data%>% select(feature.groups,features.plot) %>% arrange(feature.groups,desc(features.plot)) %>% unique() %>%\n",
    "                    ungroup() %>% mutate(gene.order=as.numeric(forcats::fct_rev(features.plot)), abs.order=seq_along(along.with = features.plot)) -> sorted_genes_last\n",
    "                if(any(!p$data$features.plot%in%unique(unlist(gene.list), recursive=T))){stop()}\n",
    "                all.genes <- unique(unlist(gene.list, recursive=T))\n",
    "                all.genes.df <- lapply(names(gene.list), function(x){data.frame(features.plot=gene.list[[x]], feature.groups=x)})%>%do.call(what='rbind')\n",
    "                all.genes.df\n",
    "                if(any(!p$data$features.plot%in%unique(all.genes.df$features.plot))){stop()}\n",
    "                sorted_genes_last <- merge(all.genes.df, sorted_genes_last, by=c('features.plot','feature.groups'), all=T)%>%mutate(expressed=!is.na(abs.order)) %>% arrange(abs.order)\n",
    "                write.csv(sorted_genes_last, glue::glue('{save.dir}{name}.genelist.csv'),row.names=F)\n",
    "                \n",
    "            }\n",
    "        }\n",
    "    }\n",
    "\n",
    "    return(ps)\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5b9b950-6bb6-4b4f-91b5-05b8106a72fc",
   "metadata": {},
   "source": [
    "# Dotplots"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4b9fb989-1f0d-4e49-beec-51aafc7072eb",
   "metadata": {},
   "source": [
    "## Figure PC tests"
   ]
  },
  {
   "cell_type": "raw",
   "id": "8f9b5d6c-5c56-4673-b79f-bc297c0550bb",
   "metadata": {},
   "source": [
    "fig(7,7)\n",
    "ggplot(filt1, aes(y=-log10(Adjusted.P.value), \n",
    "                   x=Fold.changes.of.average.expression.in.log.scale,\n",
    "                   alpha=in.selection2,\n",
    "                   fill=in.selection2)) + geom_point(shape=21) + facet_wrap(~Subtype, scales='free')\n",
    "ggplot(filt1, aes(y=-log10(Adjusted.P.value), \n",
    "                   x=Fold.changes.of.average.expression.in.log.scale,\n",
    "                   alpha=in.selection3,\n",
    "                   fill=in.selection3)) + geom_point(shape=21) + facet_wrap(~group, scales='free')\n",
    "\n",
    "# genes were selected initially by pval\n",
    "\n",
    "fig(7,7)\n",
    "ggplot(filt1, aes(y=-log10(Adjusted.P.value), \n",
    "                   x=Fold.changes.of.average.expression.in.log.scale,\n",
    "                   alpha=all.ok,\n",
    "                   fill=all.ok)) + geom_point(shape=21) + facet_wrap(~group, scales='free')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "251c5097-2869-4403-ba72-77157b7b30b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "subtype.labels <- c(\n",
    "    'AV NKX2-1 LMO1' = 'AV',\n",
    "    'AV NKX2-1 NKX6-2' = 'AV',\n",
    "    'GE RG NKX2-1 DLK1' = 'GE',\n",
    "    'GE RG NKX2-1 OLIG1' = 'GE',\n",
    "    'PC FGF17' = 'RPC (PC FGF17)',\n",
    "    'PC RSPO3' = 'Hem (PC RSPO3)',\n",
    "    'PC SFRP2' = 'Antihem (PC SFRP2)',\n",
    "    'PC TCF7L2' = 'ZLI (PC TCFL2)',\n",
    "    'PC TTR' = 'Hem CPe (PC TTR)'\n",
    ")\n",
    "filt1 <- pat.de\n",
    "filt1$group <- plyr::mapvalues(x = filt1$Subtype, from = names(subtype.labels), to=subtype.labels)\n",
    "filt1$group <- factor(filt1$group, names(nicola.selection3))\n",
    "\n",
    "filt1 <- filt1%>%group_by(group) %>% mutate(in.selection3 = gene%in%nicola.selection3[[unique(group)]]) %>% ungroup()\n",
    "filt1 <- filt1%>%group_by(Subtype) %>% mutate(in.selection2 = gene%in%nicola.selection2[[unique(Subtype)]]) %>% ungroup()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 484,
   "id": "1eaf5164-cb59-418d-a70c-2680270cd9e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "nicola2.labels <- c(\n",
    "    'AV NKX2-1 LMO1' = 'AV',\n",
    "    'AV NKX2-1 NKX6-2' = 'AV',\n",
    "    'GE RG NKX2-1 DLK1' = 'GE',\n",
    "    'GE RG NKX2-1 OLIG1' = 'GE',\n",
    "    'RPC (PC FGF17)' = 'RPC (PC FGF17)',\n",
    "    'Hem (PC RSPO3)' = 'Hem (PC RSPO3)',\n",
    "    'Antihem (PC SFRP2)' = 'Antihem (PC SFRP2)',\n",
    "    'ZLI (PC TCFL2)' = 'ZLI (PC TCFL2)',\n",
    "    'Hem CPe (PC TTR)' = 'Hem CPe (PC TTR)'\n",
    ")\n",
    "\n",
    "lapply(unique(nicola2.labels),\n",
    "       function(x){\n",
    "           unique(unlist(nicola.selection2[names(nicola2.labels)[nicola2.labels==x]], recursive=T))\n",
    "       }) %>% setNames(nm=unique(nicola2.labels)) -> nicola.selection2.grouped\n",
    "nicola.selection2.grouped <- nicola.selection2.grouped[names(nicola.selection3)]\n",
    "\n",
    "all.regional <- unlist(nescreg.list.no_trg, recursive=T)%>%unique()\n",
    "all.regional[lapply(nescreg.list.no_trg, function(x){(all.regional%in%x)%>%setNames(all.regional)})%>%data.frame %>%apply(1,sum)<=2] -> max2.regional\n",
    "lapply(nescreg.list.no_trg, function(x){x[x%in%max2.regional]}) %>%setNames(names(nescreg.list.no_trg)) -> regional.max2.lists\n",
    "\n",
    "filt1.tests <- list(\n",
    "    PCmarkers.NicolaTableSelection=nicola.selection2.grouped,\n",
    "    'RegionalMarkers' = regional.max2.lists)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 485,
   "id": "54ad4cb4-6b19-4e44-b86d-260547502799",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$PCmarkers.NicolaTableSelection</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>RPC (PC FGF17)</dt><dd>TRUE</dd><dt>AV</dt><dd>TRUE</dd><dt>GE</dt><dd>TRUE</dd><dt>Hem (PC RSPO3)</dt><dd>TRUE</dd><dt>Hem CPe (PC TTR)</dt><dd>TRUE</dd><dt>Antihem (PC SFRP2)</dt><dd>TRUE</dd><dt>ZLI (PC TCFL2)</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$RegionalMarkers</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>FR</dt><dd>TRUE</dd><dt>MS</dt><dd>TRUE</dd><dt>OC</dt><dd>TRUE</dd><dt>Tem</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$PCmarkers.NicolaTableSelection] \\begin{description*}\n",
       "\\item[RPC (PC FGF17)] TRUE\n",
       "\\item[AV] TRUE\n",
       "\\item[GE] TRUE\n",
       "\\item[Hem (PC RSPO3)] TRUE\n",
       "\\item[Hem CPe (PC TTR)] TRUE\n",
       "\\item[Antihem (PC SFRP2)] TRUE\n",
       "\\item[ZLI (PC TCFL2)] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$RegionalMarkers] \\begin{description*}\n",
       "\\item[FR] TRUE\n",
       "\\item[MS] TRUE\n",
       "\\item[OC] TRUE\n",
       "\\item[Tem] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$PCmarkers.NicolaTableSelection\n",
       ":   RPC (PC FGF17)\n",
       ":   TRUEAV\n",
       ":   TRUEGE\n",
       ":   TRUEHem (PC RSPO3)\n",
       ":   TRUEHem CPe (PC TTR)\n",
       ":   TRUEAntihem (PC SFRP2)\n",
       ":   TRUEZLI (PC TCFL2)\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "$RegionalMarkers\n",
       ":   FR\n",
       ":   TRUEMS\n",
       ":   TRUEOC\n",
       ":   TRUETem\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$PCmarkers.NicolaTableSelection\n",
       "    RPC (PC FGF17)                 AV                 GE     Hem (PC RSPO3) \n",
       "              TRUE               TRUE               TRUE               TRUE \n",
       "  Hem CPe (PC TTR) Antihem (PC SFRP2)     ZLI (PC TCFL2) \n",
       "              TRUE               TRUE               TRUE \n",
       "\n",
       "$RegionalMarkers\n",
       "  FR   MS   OC  Tem \n",
       "TRUE TRUE TRUE TRUE \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "is.unique <- function(l){\n",
    "    length(l) == length(unique(l))\n",
    "}\n",
    "are.unique <- function(l){\n",
    "    sapply(l, is.unique)\n",
    "}\n",
    "\n",
    "lapply(filt1.tests, are.unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 486,
   "id": "59c0e2ca-9129-431f-bf2b-1f7b9fea5c54",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 73 shown): H2AFX, C1H1orf61, C13H2orf40, C6H5orf49, H2AFJ, LOC106997086, KIAA1257, LOC106997082, RPL3.1, LOC100499503”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n",
      "[1] TRUE\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 80 shown): H2AFX, C1H1orf61, C13H2orf40, C6H5orf49, H2AFJ, LOC106997086, KIAA1257, LOC106997082, RPL3.1, LOC100499503”\n"
     ]
    }
   ],
   "source": [
    "topns <- c(15,20,25,30,35,45)\n",
    "topns <- c(25)\n",
    "# unsupervised using expression ranking\n",
    "filt1.ranked <- unique(unlist(filt1.tests, recursive=T))\n",
    "filt1.ranked <- FetchData(seu, filt1.ranked) %>%colSums%>%sort(decreasing = T)%>%names()\n",
    "filt1.tops <- list()\n",
    "for (top.n in topns){\n",
    "\n",
    "    lapply(filt1.tests, function(x){\n",
    "        lapply(x, function(y){\n",
    "            filt1.ranked[filt1.ranked%in%y]->z\n",
    "            print(is.unique(z))\n",
    "            if(length(z)>top.n){z[1:top.n]}else{z}\n",
    "            \n",
    "        })%>%setNames(nm=names(x))\n",
    "    })%>%setNames(nm=paste0(names(filt1.tests), '.topExpressed', as.character(top.n))) -> toplists\n",
    "    \n",
    "    filt1.tops <- c(filt1.tops, toplists)\n",
    "}\n",
    "filt1.tests <- c(filt1.tests, filt1.tops)\n",
    "\n",
    "# unsupervised lists from pcs\n",
    "for (top.n in topns){\n",
    "    tag <- glue::glue('PCtable.topn{top.n}')\n",
    "    this.filt <- filt1\n",
    "\n",
    "    filter.de.results(this.filt, 'Adjusted.P.value', 'Ratio.fold.changes', split.by = 'group', top.n=top.n) -> this.filt\n",
    "    \n",
    "    filt.list <- with(subset(this.filt, all.ok), {split(gene, group)}%>%lapply(unique))\n",
    "\n",
    "    filt1.tests[[tag]] <- filt.list\n",
    "}\n",
    "\n",
    "\n",
    "# set ordering in all lists to follow maximal sample of expression in early radial glia\n",
    "unique(unlist(filt1.tests)) -> g\n",
    "d <- DotPlot(subset(seu, anno_cluster_fct == 'RG early'), \n",
    "             features = g,\n",
    "             group.by='cell_origin')\n",
    "# first group to obtain max sample per gene\n",
    "d$data %>% group_by(features.plot) %>% summarise(ms=id[which.max(pct.exp)], exp=max(pct.exp)) %>%\n",
    "# then group by maxsample\n",
    "group_by(ms) %>% arrange(ms, desc(exp)) %>% ungroup()%>% select(features.plot) %>%\n",
    "apply(FUN=as.character, 1) %>% rev() -> filt1.glevels.all\n",
    "\n",
    "filt1.glevels <- list()\n",
    "\n",
    "for (l in names(filt1.tests)){\n",
    "    \n",
    "    filt1.glevels[[l]] <- filt1.glevels.all[filt1.glevels.all %in% unique(unlist(filt1.tests[[l]]))]\n",
    "    filt1.tests[[l]] <- lapply(filt1.tests[[l]], function(x){\n",
    "        x[order(match(x,filt1.glevels[[l]]))]\n",
    "    })\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 487,
   "id": "e123860f-f490-4832-a150-2a5fc3c3c575",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$PCmarkers.NicolaTableSelection</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>RPC (PC FGF17)</dt><dd>TRUE</dd><dt>AV</dt><dd>TRUE</dd><dt>GE</dt><dd>TRUE</dd><dt>Hem (PC RSPO3)</dt><dd>TRUE</dd><dt>Hem CPe (PC TTR)</dt><dd>TRUE</dd><dt>Antihem (PC SFRP2)</dt><dd>TRUE</dd><dt>ZLI (PC TCFL2)</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$RegionalMarkers</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>FR</dt><dd>TRUE</dd><dt>MS</dt><dd>TRUE</dd><dt>OC</dt><dd>TRUE</dd><dt>Tem</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$PCmarkers.NicolaTableSelection.topExpressed25</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>RPC (PC FGF17)</dt><dd>TRUE</dd><dt>AV</dt><dd>TRUE</dd><dt>GE</dt><dd>TRUE</dd><dt>Hem (PC RSPO3)</dt><dd>TRUE</dd><dt>Hem CPe (PC TTR)</dt><dd>TRUE</dd><dt>Antihem (PC SFRP2)</dt><dd>TRUE</dd><dt>ZLI (PC TCFL2)</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$RegionalMarkers.topExpressed25</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>FR</dt><dd>TRUE</dd><dt>MS</dt><dd>TRUE</dd><dt>OC</dt><dd>TRUE</dd><dt>Tem</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "\t<dt>$PCtable.topn25</dt>\n",
       "\t\t<dd><style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>RPC (PC FGF17)</dt><dd>TRUE</dd><dt>AV</dt><dd>TRUE</dd><dt>GE</dt><dd>TRUE</dd><dt>Hem (PC RSPO3)</dt><dd>TRUE</dd><dt>Hem CPe (PC TTR)</dt><dd>TRUE</dd><dt>Antihem (PC SFRP2)</dt><dd>TRUE</dd><dt>ZLI (PC TCFL2)</dt><dd>TRUE</dd></dl>\n",
       "</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$PCmarkers.NicolaTableSelection] \\begin{description*}\n",
       "\\item[RPC (PC FGF17)] TRUE\n",
       "\\item[AV] TRUE\n",
       "\\item[GE] TRUE\n",
       "\\item[Hem (PC RSPO3)] TRUE\n",
       "\\item[Hem CPe (PC TTR)] TRUE\n",
       "\\item[Antihem (PC SFRP2)] TRUE\n",
       "\\item[ZLI (PC TCFL2)] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$RegionalMarkers] \\begin{description*}\n",
       "\\item[FR] TRUE\n",
       "\\item[MS] TRUE\n",
       "\\item[OC] TRUE\n",
       "\\item[Tem] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$PCmarkers.NicolaTableSelection.topExpressed25] \\begin{description*}\n",
       "\\item[RPC (PC FGF17)] TRUE\n",
       "\\item[AV] TRUE\n",
       "\\item[GE] TRUE\n",
       "\\item[Hem (PC RSPO3)] TRUE\n",
       "\\item[Hem CPe (PC TTR)] TRUE\n",
       "\\item[Antihem (PC SFRP2)] TRUE\n",
       "\\item[ZLI (PC TCFL2)] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$RegionalMarkers.topExpressed25] \\begin{description*}\n",
       "\\item[FR] TRUE\n",
       "\\item[MS] TRUE\n",
       "\\item[OC] TRUE\n",
       "\\item[Tem] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\item[\\$PCtable.topn25] \\begin{description*}\n",
       "\\item[RPC (PC FGF17)] TRUE\n",
       "\\item[AV] TRUE\n",
       "\\item[GE] TRUE\n",
       "\\item[Hem (PC RSPO3)] TRUE\n",
       "\\item[Hem CPe (PC TTR)] TRUE\n",
       "\\item[Antihem (PC SFRP2)] TRUE\n",
       "\\item[ZLI (PC TCFL2)] TRUE\n",
       "\\end{description*}\n",
       "\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$PCmarkers.NicolaTableSelection\n",
       ":   RPC (PC FGF17)\n",
       ":   TRUEAV\n",
       ":   TRUEGE\n",
       ":   TRUEHem (PC RSPO3)\n",
       ":   TRUEHem CPe (PC TTR)\n",
       ":   TRUEAntihem (PC SFRP2)\n",
       ":   TRUEZLI (PC TCFL2)\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "$RegionalMarkers\n",
       ":   FR\n",
       ":   TRUEMS\n",
       ":   TRUEOC\n",
       ":   TRUETem\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "$PCmarkers.NicolaTableSelection.topExpressed25\n",
       ":   RPC (PC FGF17)\n",
       ":   TRUEAV\n",
       ":   TRUEGE\n",
       ":   TRUEHem (PC RSPO3)\n",
       ":   TRUEHem CPe (PC TTR)\n",
       ":   TRUEAntihem (PC SFRP2)\n",
       ":   TRUEZLI (PC TCFL2)\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "$RegionalMarkers.topExpressed25\n",
       ":   FR\n",
       ":   TRUEMS\n",
       ":   TRUEOC\n",
       ":   TRUETem\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "$PCtable.topn25\n",
       ":   RPC (PC FGF17)\n",
       ":   TRUEAV\n",
       ":   TRUEGE\n",
       ":   TRUEHem (PC RSPO3)\n",
       ":   TRUEHem CPe (PC TTR)\n",
       ":   TRUEAntihem (PC SFRP2)\n",
       ":   TRUEZLI (PC TCFL2)\n",
       ":   TRUE\n",
       "\n",
       "\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$PCmarkers.NicolaTableSelection\n",
       "    RPC (PC FGF17)                 AV                 GE     Hem (PC RSPO3) \n",
       "              TRUE               TRUE               TRUE               TRUE \n",
       "  Hem CPe (PC TTR) Antihem (PC SFRP2)     ZLI (PC TCFL2) \n",
       "              TRUE               TRUE               TRUE \n",
       "\n",
       "$RegionalMarkers\n",
       "  FR   MS   OC  Tem \n",
       "TRUE TRUE TRUE TRUE \n",
       "\n",
       "$PCmarkers.NicolaTableSelection.topExpressed25\n",
       "    RPC (PC FGF17)                 AV                 GE     Hem (PC RSPO3) \n",
       "              TRUE               TRUE               TRUE               TRUE \n",
       "  Hem CPe (PC TTR) Antihem (PC SFRP2)     ZLI (PC TCFL2) \n",
       "              TRUE               TRUE               TRUE \n",
       "\n",
       "$RegionalMarkers.topExpressed25\n",
       "  FR   MS   OC  Tem \n",
       "TRUE TRUE TRUE TRUE \n",
       "\n",
       "$PCtable.topn25\n",
       "    RPC (PC FGF17)                 AV                 GE     Hem (PC RSPO3) \n",
       "              TRUE               TRUE               TRUE               TRUE \n",
       "  Hem CPe (PC TTR) Antihem (PC SFRP2)     ZLI (PC TCFL2) \n",
       "              TRUE               TRUE               TRUE \n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "lapply(filt1.tests, are.unique)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 488,
   "id": "1d494b27-ec1d-42d6-8d15-deb6ad3b25cc",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(fig_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10' already exists”\n",
      "PCmarkers.NicolaTableSelection\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10' already exists”\n",
      "PCmarkers.NicolaTableSelection\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 (41.67%) repeated genes in RPC (PC FGF17) list\n",
      "33 (48.53%) repeated genes in AV list\n",
      "26 (43.33%) repeated genes in GE list\n",
      "15 (33.33%) repeated genes in Hem (PC RSPO3) list\n",
      "4 (10.53%) repeated genes in Hem CPe (PC TTR) list\n",
      "8 (32%) repeated genes in Antihem (PC SFRP2) list\n",
      "6 (18.18%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found: H2AFX, C1H1orf61, C13H2orf40, C6H5orf49, H2AFJ”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 242 -> 196\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RegionalMarkers\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10' already exists”\n",
      "RegionalMarkers\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 (9.16%) repeated genes in FR list\n",
      "19 (15.45%) repeated genes in MS list\n",
      "23 (19.66%) repeated genes in OC list\n",
      "7 (31.82%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 68 shown): LOC106997086, KIAA1257, LOC106997082, RPL3.1, LOC100499503, LOC708725, LOC109910387, LOC114674534, HIST1H1E, LOC696005”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 430 -> 375\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PCmarkers.NicolaTableSelection.topExpressed25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10' already exists”\n",
      "PCmarkers.NicolaTableSelection.topExpressed25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 (32%) repeated genes in RPC (PC FGF17) list\n",
      "12 (48%) repeated genes in AV list\n",
      "10 (40%) repeated genes in GE list\n",
      "10 (40%) repeated genes in Hem (PC RSPO3) list\n",
      "3 (12%) repeated genes in Hem CPe (PC TTR) list\n",
      "6 (24%) repeated genes in Antihem (PC SFRP2) list\n",
      "5 (20%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 144 -> 138\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RegionalMarkers.topExpressed25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10' already exists”\n",
      "RegionalMarkers.topExpressed25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (8%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "4 (16%) repeated genes in OC list\n",
      "4 (18.18%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 92 -> 87\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PCtable.topn25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10' already exists”\n",
      "PCtable.topn25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 (28%) repeated genes in RPC (PC FGF17) list\n",
      "9 (37.5%) repeated genes in AV list\n",
      "6 (31.58%) repeated genes in GE list\n",
      "1 (4%) repeated genes in Hem (PC RSPO3) list\n",
      "0 (0%) repeated genes in Hem CPe (PC TTR) list\n",
      "4 (16%) repeated genes in Antihem (PC SFRP2) list\n",
      "4 (16%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 13 shown): LOC114676301, LOC711450, LOC114674534, LOC106993104, LOC716075, LOC106998526, LOC107000288, LOC106998839, C13H2orf40, C6H5orf49”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 138 -> 111\n",
      "[1] \"#######\"\n",
      "[1] \"RG early\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(div_fig_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early' already exists”\n",
      "PCmarkers.NicolaTableSelection\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early' already exists”\n",
      "PCmarkers.NicolaTableSelection\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 (41.67%) repeated genes in RPC (PC FGF17) list\n",
      "33 (48.53%) repeated genes in AV list\n",
      "26 (43.33%) repeated genes in GE list\n",
      "15 (33.33%) repeated genes in Hem (PC RSPO3) list\n",
      "4 (10.53%) repeated genes in Hem CPe (PC TTR) list\n",
      "8 (32%) repeated genes in Antihem (PC SFRP2) list\n",
      "6 (18.18%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found: H2AFX, C1H1orf61, C13H2orf40, C6H5orf49, H2AFJ”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 242 -> 143\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RegionalMarkers\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early' already exists”\n",
      "RegionalMarkers\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 (9.16%) repeated genes in FR list\n",
      "19 (15.45%) repeated genes in MS list\n",
      "23 (19.66%) repeated genes in OC list\n",
      "7 (31.82%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 68 shown): LOC106997086, KIAA1257, LOC106997082, RPL3.1, LOC100499503, LOC708725, LOC109910387, LOC114674534, HIST1H1E, LOC696005”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 430 -> 303\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PCmarkers.NicolaTableSelection.topExpressed25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early' already exists”\n",
      "PCmarkers.NicolaTableSelection.topExpressed25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 (32%) repeated genes in RPC (PC FGF17) list\n",
      "12 (48%) repeated genes in AV list\n",
      "10 (40%) repeated genes in GE list\n",
      "10 (40%) repeated genes in Hem (PC RSPO3) list\n",
      "3 (12%) repeated genes in Hem CPe (PC TTR) list\n",
      "6 (24%) repeated genes in Antihem (PC SFRP2) list\n",
      "5 (20%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 144 -> 115\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RegionalMarkers.topExpressed25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early' already exists”\n",
      "RegionalMarkers.topExpressed25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (8%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "4 (16%) repeated genes in OC list\n",
      "4 (18.18%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 92 -> 82\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PCtable.topn25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early' already exists”\n",
      "PCtable.topn25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 (28%) repeated genes in RPC (PC FGF17) list\n",
      "9 (37.5%) repeated genes in AV list\n",
      "6 (31.58%) repeated genes in GE list\n",
      "1 (4%) repeated genes in Hem (PC RSPO3) list\n",
      "0 (0%) repeated genes in Hem CPe (PC TTR) list\n",
      "4 (16%) repeated genes in Antihem (PC SFRP2) list\n",
      "4 (16%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 13 shown): LOC114676301, LOC711450, LOC114674534, LOC106993104, LOC716075, LOC106998526, LOC107000288, LOC106998839, C13H2orf40, C6H5orf49”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 138 -> 83\n",
      "[1] \"#######\"\n",
      "[1] \"RG late\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(div_fig_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late' already exists”\n",
      "PCmarkers.NicolaTableSelection\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late' already exists”\n",
      "PCmarkers.NicolaTableSelection\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 (41.67%) repeated genes in RPC (PC FGF17) list\n",
      "33 (48.53%) repeated genes in AV list\n",
      "26 (43.33%) repeated genes in GE list\n",
      "15 (33.33%) repeated genes in Hem (PC RSPO3) list\n",
      "4 (10.53%) repeated genes in Hem CPe (PC TTR) list\n",
      "8 (32%) repeated genes in Antihem (PC SFRP2) list\n",
      "6 (18.18%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found: H2AFX, C1H1orf61, C13H2orf40, C6H5orf49, H2AFJ”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 242 -> 115\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RegionalMarkers\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late' already exists”\n",
      "RegionalMarkers\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 (9.16%) repeated genes in FR list\n",
      "19 (15.45%) repeated genes in MS list\n",
      "23 (19.66%) repeated genes in OC list\n",
      "7 (31.82%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 68 shown): LOC106997086, KIAA1257, LOC106997082, RPL3.1, LOC100499503, LOC708725, LOC109910387, LOC114674534, HIST1H1E, LOC696005”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 430 -> 229\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PCmarkers.NicolaTableSelection.topExpressed25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late' already exists”\n",
      "PCmarkers.NicolaTableSelection.topExpressed25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8 (32%) repeated genes in RPC (PC FGF17) list\n",
      "12 (48%) repeated genes in AV list\n",
      "10 (40%) repeated genes in GE list\n",
      "10 (40%) repeated genes in Hem (PC RSPO3) list\n",
      "3 (12%) repeated genes in Hem CPe (PC TTR) list\n",
      "6 (24%) repeated genes in Antihem (PC SFRP2) list\n",
      "5 (20%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 144 -> 99\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "RegionalMarkers.topExpressed25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late' already exists”\n",
      "RegionalMarkers.topExpressed25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (8%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "4 (16%) repeated genes in OC list\n",
      "4 (18.18%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 92 -> 78\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PCtable.topn25\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late' already exists”\n",
      "PCtable.topn25\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7 (28%) repeated genes in RPC (PC FGF17) list\n",
      "9 (37.5%) repeated genes in AV list\n",
      "6 (31.58%) repeated genes in GE list\n",
      "1 (4%) repeated genes in Hem (PC RSPO3) list\n",
      "0 (0%) repeated genes in Hem CPe (PC TTR) list\n",
      "4 (16%) repeated genes in Antihem (PC SFRP2) list\n",
      "4 (16%) repeated genes in ZLI (PC TCFL2) list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“The following requested variables were not found (10 out of 13 shown): LOC114676301, LOC711450, LOC114674534, LOC106993104, LOC716075, LOC106998526, LOC107000288, LOC106998839, C13H2orf40, C6H5orf49”\n",
      "Requiring 10% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 138 -> 68\n"
     ]
    }
   ],
   "source": [
    "# fig_tag <- 'tests'\n",
    "# minpcts <-  c(5,10,15)\n",
    "\n",
    "fig_tag <- 'selection'\n",
    "minpcts <-  c(10)\n",
    "\n",
    "for (min.pct in minpcts){\n",
    "\n",
    "    fig_dir <- paste0(original.dir, 'figure_', fig_tag, '/minpct', as.character(min.pct), '/')\n",
    "    dir.create(fig_dir, recursive = T)\n",
    "\n",
    "    # without clustering\n",
    "    for (l in names(filt1.tests)){\n",
    "        message(l)\n",
    "        dotplot.save(seu = seu, gene.list = filt1.tests[[l]], name = l, with_clustering = F,\n",
    "                     pct.min = min.pct, genelevels=filt1.glevels,\n",
    "                     save.dir = fig_dir)\n",
    "        # stop()\n",
    "    }\n",
    "    # without clustering\n",
    "    for (ct in unique(seu$anno_cluster_fct)){\n",
    "        if (!ct%in%c('RG early', 'RG late')){next()}\n",
    "        print('#######')\n",
    "        print(ct)\n",
    "        print('#######')\n",
    "\n",
    "        if (! ct %in% names(ct_seus)){\n",
    "            ct_seus[[ct]] <- ScaleData(subset(seu, anno_cluster_fct == ct), verbose=T)\n",
    "        } \n",
    "        ct_seu <- ct_seus[[ct]]\n",
    "\n",
    "        ct <- gsub(x=ct, pattern='/', replacement=' ' )\n",
    "        div_fig_dir <- paste0(fig_dir, 'per_celltype/', ct, '/')\n",
    "        dir.create(div_fig_dir, recursive = T)\n",
    "\n",
    "        for (l in names(filt1.tests)){\n",
    "            message(l)\n",
    "\n",
    "            ps <- dotplot.save(\n",
    "                seu = ct_seu, gene.list = filt1.tests[[l]], name = l,\n",
    "                with_clustering = F, save.dir = div_fig_dir, min.width=2.7,\n",
    "                pct.min = min.pct, genelevels=filt1.glevels,\n",
    "            )\n",
    "        }\n",
    "        # stop()\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 490,
   "id": "4e34bcd1-42e8-4f84-ba03-1656ac9ecbd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/PCmarkers.NicolaTableSelection.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/PCmarkers.NicolaTableSelection.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/PCtable.topn25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/PCmarkers.NicolaTableSelection.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/PCmarkers.NicolaTableSelection.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/PCtable.topn25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/RegionalMarkers.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/RegionalMarkers.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/PCmarkers.NicolaTableSelection.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/PCmarkers.NicolaTableSelection.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/PCtable.topn25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/RegionalMarkers.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/RegionalMarkers.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/RegionalMarkers.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/RegionalMarkers.topExpressed25.genelist.csv\"\n"
     ]
    }
   ],
   "source": [
    "fig.files <- list.files('~/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10', all.files = T, full.names = T, recursive = T) %>%grep(pattern='.csv', fixed=T, value=T)\n",
    "# fig.files\n",
    "# names(filt1.tests)\n",
    "tbls <- list()\n",
    "for (ff in fig.files){\n",
    "    print(ff)\n",
    "    d <- read.csv(ff)%>%rename(gene=features.plot, gene.group=feature.groups)\n",
    "    \n",
    "    if (grepl('per_celltype', ff)){\n",
    "        ct <- basename(dirname(ff))\n",
    "        ori.table <- gsub('.genelist.csv', paste0('.',ct), basename(ff), fixed=T)\n",
    "    }else{\n",
    "        ori.table <- gsub('.genelist.csv', '', basename(ff), fixed=T)\n",
    "        ct <- 'AllCellTypes'\n",
    "    }\n",
    "    # d[paste0(ori.table, '.order')] <- d$gene.order\n",
    "    d$gene.order<-NULL\n",
    "    d$ct <- ct\n",
    "    # d[paste0(ori.table, '.inlist')] <- !is.na(d$abs.order)\n",
    "    # d$abs.order<-NULL\n",
    "    # d <- merge(data.frame(gene=filt1.glevels[[ori.table]]), d, by='gene', all=T)\n",
    "    # d[paste0(ori.table, '.expressed')] <- !is.na(d[,3]) \n",
    "    tbls[[ori.table]] <- d#[,c(2,1,4,3)]\n",
    "}\n",
    "\n",
    "# print(names(tbls))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "id": "633c9b0b-5b29-4816-b9d1-83f097746068",
   "metadata": {},
   "outputs": [],
   "source": [
    "tbls[c(\"PCmarkers.NicolaTableSelection\",\"PCmarkers.NicolaTableSelection.RG early\",\"PCmarkers.NicolaTableSelection.RG late\")]%>%do.call(what='rbind') -> exptbl.pcm1\n",
    "merge(\n",
    "    exptbl.pcm1 %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='expressed'), exptbl.pcm1 %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='abs.order'),\n",
    "    by=c('gene.group', 'gene'), all=T, suffixes=c('.ExpressedInAtLeast10%In1Donor', '.PlotOrder')) %>% arrange(AllCellTypes.PlotOrder)  -> exptbl.pcm1\n",
    "\n",
    "\n",
    "tbls[c(\"PCmarkers.NicolaTableSelection.topExpressed25\",\"PCmarkers.NicolaTableSelection.topExpressed25.RG early\",\"PCmarkers.NicolaTableSelection.topExpressed25.RG late\")]%>%do.call(what='rbind') -> exptbl.pcm1.top\n",
    "merge(\n",
    "    exptbl.pcm1.top %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='expressed'), exptbl.pcm1.top %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='abs.order'),\n",
    "    by=c('gene.group', 'gene'), all=T, suffixes=c('.ExpressedInAtLeast10%In1Donor', '.PlotOrder')) %>% arrange(AllCellTypes.PlotOrder)  -> exptbl.pcm1.top\n",
    "\n",
    "\n",
    "tbls[c(\"PCtable.topn25\",\"PCtable.topn25.RG early\",\"PCtable.topn25.RG late\")]%>%do.call(what='rbind') -> exptbl.pcm2\n",
    "merge(\n",
    "    exptbl.pcm2 %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='expressed'), exptbl.pcm2 %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='abs.order'),\n",
    "    by=c('gene.group', 'gene'), all=T, suffixes=c('.ExpressedInAtLeast10%In1Donor', '.PlotOrder')) %>% arrange(AllCellTypes.PlotOrder)  -> exptbl.pcm2\n",
    "\n",
    "\n",
    "tbls[c(\"RegionalMarkers\",\"RegionalMarkers.RG early\",\"RegionalMarkers.RG late\")]%>%do.call(what='rbind') -> exptbl.rm\n",
    "merge(\n",
    "    exptbl.rm %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='expressed'), exptbl.rm %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='abs.order'),\n",
    "    by=c('gene.group', 'gene'), all=T, suffixes=c('.ExpressedInAtLeast10%In1Donor', '.PlotOrder')) %>% arrange(AllCellTypes.PlotOrder)  -> exptbl.rm\n",
    "\n",
    "\n",
    "tbls[c(\"RegionalMarkers.topExpressed25\",\"RegionalMarkers.topExpressed25.RG early\",\"RegionalMarkers.topExpressed25.RG late\")]%>%do.call(what='rbind') -> exptbl.rm.top\n",
    "merge(\n",
    "    exptbl.rm.top %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='expressed'), exptbl.rm.top %>% reshape2::dcast(formula='gene.group+gene~ct', value.var='abs.order'),\n",
    "    by=c('gene.group', 'gene'), all=T, suffixes=c('.ExpressedInAtLeast10%In1Donor', '.PlotOrder')) %>% arrange(AllCellTypes.PlotOrder)  -> exptbl.rm.top\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 500,
   "id": "c98690de-3aa1-4888-9c3e-b839f7895ca3",
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs <- list(\n",
    "    # PCMarkersNicola=exptbl.pcm1,\n",
    "    # PCMarkersNicolaTopExpressed=exptbl.pcm1.top,\n",
    "    PCMarkersTop25=exptbl.pcm2,\n",
    "    RegionalMarkers=exptbl.rm,\n",
    "    RegionalMarkersTopExpressed=exptbl.rm.top\n",
    ")\n",
    "openxlsx::write.xlsx(x = dfs, file = './gene_lists_used.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 430,
   "id": "2554c99f-52c9-4def-8f7e-bd2f3743a56b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/PCmarkers.NicolaTableSelection.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/PCmarkers.NicolaTableSelection.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/PCtable.topn25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/PCmarkers.NicolaTableSelection.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/PCmarkers.NicolaTableSelection.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/PCtable.topn25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/RegionalMarkers.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG early/RegionalMarkers.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/PCmarkers.NicolaTableSelection.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/PCmarkers.NicolaTableSelection.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/PCtable.topn25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/RegionalMarkers.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/per_celltype/RG late/RegionalMarkers.topExpressed25.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/RegionalMarkers.genelist.csv\"\n",
      "[1] \"/users/genomics/xoel/codebases/ASD_IPSC_sept/plots/dotplots_2024-03-06/not_scaled/figure_selection/minpct10/RegionalMarkers.topExpressed25.genelist.csv\"\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 537 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>gene.group</th><th scope=col>expressed</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>JUND    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>CDT1    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>BNIP3   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>PDLIM1  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>SSBP4   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>ZNF775  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>MEIS3   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>8</th><td>NKAIN3  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>FABP7   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>10</th><td>SAT1    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>GAS6    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>12</th><td>ADAMTS19</td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>13</th><td>SFRP1   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>14</th><td>METRN   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>15</th><td>TPM1    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>16</th><td>HMGA1   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>17</th><td>IGFBP5  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>18</th><td>PTN     </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>19</th><td>CD81    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>20</th><td>ARL4C   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>21</th><td>EIF5A   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>22</th><td>PRXL2A  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>23</th><td>RPS4Y1  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>24</th><td>CYBA    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>25</th><td>PSAT1   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>26</th><td>COL2A1  </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>27</th><td>SPRY1   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>28</th><td>IDH2    </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>29</th><td>FBLN1   </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>30</th><td>ID2     </td><td>FR</td><td>TRUE</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>508</th><td>MAPK11  </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>509</th><td>NDRG2   </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>510</th><td>PNOC    </td><td>Tem</td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>511</th><td>PPFIBP2 </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>512</th><td>RARRES2 </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>513</th><td>RBM14-c1</td><td>MS </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>514</th><td>REEP1   </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>515</th><td>RPL14.1 </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>516</th><td>RPL3.1  </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>517</th><td>RPL31.1 </td><td>MS </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>518</th><td>RPS12.1 </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>519</th><td>RPS23.1 </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>520</th><td>RPS4Y2  </td><td>MS </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>521</th><td>SEZ6L   </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>522</th><td>SLCO1C1 </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>523</th><td>SOCS2   </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>524</th><td>SP5     </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>525</th><td>SPARCL1 </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>526</th><td>SPINK2  </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>527</th><td>SYNM    </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>528</th><td>TENM2   </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>529</th><td>TEX15   </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>530</th><td>TNC     </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>531</th><td>TSHZ3   </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>532</th><td>TSPAN7  </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>533</th><td>TTR     </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>534</th><td>TUBB2B.1</td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>535</th><td>WNT5A   </td><td>OC </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>536</th><td>WNT8B   </td><td>FR </td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>537</th><td>NA      </td><td>Tem</td><td>   NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 537 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & gene & gene.group & expressed\\\\\n",
       "  & <chr> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & JUND     & FR & TRUE\\\\\n",
       "\t2 & CDT1     & FR & TRUE\\\\\n",
       "\t3 & BNIP3    & FR & TRUE\\\\\n",
       "\t4 & PDLIM1   & FR & TRUE\\\\\n",
       "\t5 & SSBP4    & FR & TRUE\\\\\n",
       "\t6 & ZNF775   & FR & TRUE\\\\\n",
       "\t7 & MEIS3    & FR & TRUE\\\\\n",
       "\t8 & NKAIN3   & FR & TRUE\\\\\n",
       "\t9 & FABP7    & FR & TRUE\\\\\n",
       "\t10 & SAT1     & FR & TRUE\\\\\n",
       "\t11 & GAS6     & FR & TRUE\\\\\n",
       "\t12 & ADAMTS19 & FR & TRUE\\\\\n",
       "\t13 & SFRP1    & FR & TRUE\\\\\n",
       "\t14 & METRN    & FR & TRUE\\\\\n",
       "\t15 & TPM1     & FR & TRUE\\\\\n",
       "\t16 & HMGA1    & FR & TRUE\\\\\n",
       "\t17 & IGFBP5   & FR & TRUE\\\\\n",
       "\t18 & PTN      & FR & TRUE\\\\\n",
       "\t19 & CD81     & FR & TRUE\\\\\n",
       "\t20 & ARL4C    & FR & TRUE\\\\\n",
       "\t21 & EIF5A    & FR & TRUE\\\\\n",
       "\t22 & PRXL2A   & FR & TRUE\\\\\n",
       "\t23 & RPS4Y1   & FR & TRUE\\\\\n",
       "\t24 & CYBA     & FR & TRUE\\\\\n",
       "\t25 & PSAT1    & FR & TRUE\\\\\n",
       "\t26 & COL2A1   & FR & TRUE\\\\\n",
       "\t27 & SPRY1    & FR & TRUE\\\\\n",
       "\t28 & IDH2     & FR & TRUE\\\\\n",
       "\t29 & FBLN1    & FR & TRUE\\\\\n",
       "\t30 & ID2      & FR & TRUE\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t508 & MAPK11   & FR  & FALSE\\\\\n",
       "\t509 & NDRG2    & FR  & FALSE\\\\\n",
       "\t510 & PNOC     & Tem & FALSE\\\\\n",
       "\t511 & PPFIBP2  & FR  & FALSE\\\\\n",
       "\t512 & RARRES2  & FR  & FALSE\\\\\n",
       "\t513 & RBM14-c1 & MS  & FALSE\\\\\n",
       "\t514 & REEP1    & FR  & FALSE\\\\\n",
       "\t515 & RPL14.1  & OC  & FALSE\\\\\n",
       "\t516 & RPL3.1   & FR  & FALSE\\\\\n",
       "\t517 & RPL31.1  & MS  & FALSE\\\\\n",
       "\t518 & RPS12.1  & OC  & FALSE\\\\\n",
       "\t519 & RPS23.1  & OC  & FALSE\\\\\n",
       "\t520 & RPS4Y2   & MS  & FALSE\\\\\n",
       "\t521 & SEZ6L    & FR  & FALSE\\\\\n",
       "\t522 & SLCO1C1  & FR  & FALSE\\\\\n",
       "\t523 & SOCS2    & FR  & FALSE\\\\\n",
       "\t524 & SP5      & OC  & FALSE\\\\\n",
       "\t525 & SPARCL1  & FR  & FALSE\\\\\n",
       "\t526 & SPINK2   & FR  & FALSE\\\\\n",
       "\t527 & SYNM     & FR  & FALSE\\\\\n",
       "\t528 & TENM2    & OC  & FALSE\\\\\n",
       "\t529 & TEX15    & OC  & FALSE\\\\\n",
       "\t530 & TNC      & FR  & FALSE\\\\\n",
       "\t531 & TSHZ3    & OC  & FALSE\\\\\n",
       "\t532 & TSPAN7   & FR  & FALSE\\\\\n",
       "\t533 & TTR      & OC  & FALSE\\\\\n",
       "\t534 & TUBB2B.1 & FR  & FALSE\\\\\n",
       "\t535 & WNT5A    & OC  & FALSE\\\\\n",
       "\t536 & WNT8B    & FR  & FALSE\\\\\n",
       "\t537 & NA       & Tem &    NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 537 × 3\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | gene.group &lt;chr&gt; | expressed &lt;lgl&gt; |\n",
       "|---|---|---|---|\n",
       "| 1 | JUND     | FR | TRUE |\n",
       "| 2 | CDT1     | FR | TRUE |\n",
       "| 3 | BNIP3    | FR | TRUE |\n",
       "| 4 | PDLIM1   | FR | TRUE |\n",
       "| 5 | SSBP4    | FR | TRUE |\n",
       "| 6 | ZNF775   | FR | TRUE |\n",
       "| 7 | MEIS3    | FR | TRUE |\n",
       "| 8 | NKAIN3   | FR | TRUE |\n",
       "| 9 | FABP7    | FR | TRUE |\n",
       "| 10 | SAT1     | FR | TRUE |\n",
       "| 11 | GAS6     | FR | TRUE |\n",
       "| 12 | ADAMTS19 | FR | TRUE |\n",
       "| 13 | SFRP1    | FR | TRUE |\n",
       "| 14 | METRN    | FR | TRUE |\n",
       "| 15 | TPM1     | FR | TRUE |\n",
       "| 16 | HMGA1    | FR | TRUE |\n",
       "| 17 | IGFBP5   | FR | TRUE |\n",
       "| 18 | PTN      | FR | TRUE |\n",
       "| 19 | CD81     | FR | TRUE |\n",
       "| 20 | ARL4C    | FR | TRUE |\n",
       "| 21 | EIF5A    | FR | TRUE |\n",
       "| 22 | PRXL2A   | FR | TRUE |\n",
       "| 23 | RPS4Y1   | FR | TRUE |\n",
       "| 24 | CYBA     | FR | TRUE |\n",
       "| 25 | PSAT1    | FR | TRUE |\n",
       "| 26 | COL2A1   | FR | TRUE |\n",
       "| 27 | SPRY1    | FR | TRUE |\n",
       "| 28 | IDH2     | FR | TRUE |\n",
       "| 29 | FBLN1    | FR | TRUE |\n",
       "| 30 | ID2      | FR | TRUE |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 508 | MAPK11   | FR  | FALSE |\n",
       "| 509 | NDRG2    | FR  | FALSE |\n",
       "| 510 | PNOC     | Tem | FALSE |\n",
       "| 511 | PPFIBP2  | FR  | FALSE |\n",
       "| 512 | RARRES2  | FR  | FALSE |\n",
       "| 513 | RBM14-c1 | MS  | FALSE |\n",
       "| 514 | REEP1    | FR  | FALSE |\n",
       "| 515 | RPL14.1  | OC  | FALSE |\n",
       "| 516 | RPL3.1   | FR  | FALSE |\n",
       "| 517 | RPL31.1  | MS  | FALSE |\n",
       "| 518 | RPS12.1  | OC  | FALSE |\n",
       "| 519 | RPS23.1  | OC  | FALSE |\n",
       "| 520 | RPS4Y2   | MS  | FALSE |\n",
       "| 521 | SEZ6L    | FR  | FALSE |\n",
       "| 522 | SLCO1C1  | FR  | FALSE |\n",
       "| 523 | SOCS2    | FR  | FALSE |\n",
       "| 524 | SP5      | OC  | FALSE |\n",
       "| 525 | SPARCL1  | FR  | FALSE |\n",
       "| 526 | SPINK2   | FR  | FALSE |\n",
       "| 527 | SYNM     | FR  | FALSE |\n",
       "| 528 | TENM2    | OC  | FALSE |\n",
       "| 529 | TEX15    | OC  | FALSE |\n",
       "| 530 | TNC      | FR  | FALSE |\n",
       "| 531 | TSHZ3    | OC  | FALSE |\n",
       "| 532 | TSPAN7   | FR  | FALSE |\n",
       "| 533 | TTR      | OC  | FALSE |\n",
       "| 534 | TUBB2B.1 | FR  | FALSE |\n",
       "| 535 | WNT5A    | OC  | FALSE |\n",
       "| 536 | WNT8B    | FR  | FALSE |\n",
       "| 537 | NA       | Tem |    NA |\n",
       "\n"
      ],
      "text/plain": [
       "    gene     gene.group expressed\n",
       "1   JUND     FR         TRUE     \n",
       "2   CDT1     FR         TRUE     \n",
       "3   BNIP3    FR         TRUE     \n",
       "4   PDLIM1   FR         TRUE     \n",
       "5   SSBP4    FR         TRUE     \n",
       "6   ZNF775   FR         TRUE     \n",
       "7   MEIS3    FR         TRUE     \n",
       "8   NKAIN3   FR         TRUE     \n",
       "9   FABP7    FR         TRUE     \n",
       "10  SAT1     FR         TRUE     \n",
       "11  GAS6     FR         TRUE     \n",
       "12  ADAMTS19 FR         TRUE     \n",
       "13  SFRP1    FR         TRUE     \n",
       "14  METRN    FR         TRUE     \n",
       "15  TPM1     FR         TRUE     \n",
       "16  HMGA1    FR         TRUE     \n",
       "17  IGFBP5   FR         TRUE     \n",
       "18  PTN      FR         TRUE     \n",
       "19  CD81     FR         TRUE     \n",
       "20  ARL4C    FR         TRUE     \n",
       "21  EIF5A    FR         TRUE     \n",
       "22  PRXL2A   FR         TRUE     \n",
       "23  RPS4Y1   FR         TRUE     \n",
       "24  CYBA     FR         TRUE     \n",
       "25  PSAT1    FR         TRUE     \n",
       "26  COL2A1   FR         TRUE     \n",
       "27  SPRY1    FR         TRUE     \n",
       "28  IDH2     FR         TRUE     \n",
       "29  FBLN1    FR         TRUE     \n",
       "30  ID2      FR         TRUE     \n",
       "⋮   ⋮        ⋮          ⋮        \n",
       "508 MAPK11   FR         FALSE    \n",
       "509 NDRG2    FR         FALSE    \n",
       "510 PNOC     Tem        FALSE    \n",
       "511 PPFIBP2  FR         FALSE    \n",
       "512 RARRES2  FR         FALSE    \n",
       "513 RBM14-c1 MS         FALSE    \n",
       "514 REEP1    FR         FALSE    \n",
       "515 RPL14.1  OC         FALSE    \n",
       "516 RPL3.1   FR         FALSE    \n",
       "517 RPL31.1  MS         FALSE    \n",
       "518 RPS12.1  OC         FALSE    \n",
       "519 RPS23.1  OC         FALSE    \n",
       "520 RPS4Y2   MS         FALSE    \n",
       "521 SEZ6L    FR         FALSE    \n",
       "522 SLCO1C1  FR         FALSE    \n",
       "523 SOCS2    FR         FALSE    \n",
       "524 SP5      OC         FALSE    \n",
       "525 SPARCL1  FR         FALSE    \n",
       "526 SPINK2   FR         FALSE    \n",
       "527 SYNM     FR         FALSE    \n",
       "528 TENM2    OC         FALSE    \n",
       "529 TEX15    OC         FALSE    \n",
       "530 TNC      FR         FALSE    \n",
       "531 TSHZ3    OC         FALSE    \n",
       "532 TSPAN7   FR         FALSE    \n",
       "533 TTR      OC         FALSE    \n",
       "534 TUBB2B.1 FR         FALSE    \n",
       "535 WNT5A    OC         FALSE    \n",
       "536 WNT8B    FR         FALSE    \n",
       "537 NA       Tem           NA    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# tbls[['RegionalMarkers']] <- merge(tbls[['RegionalMarkers']],  \n",
    "#                                    tbls[['RegionalMarkers.topExpressed25']]%>%select(-abs.order)%>%rename(in.topExpressed25=expressed), \n",
    "#                                    by=c('gene', 'gene.group'), all=T) %>% arrange(abs.order)\n",
    "# tbls[['RegionalMarkers.topExpressed25']] <- NULL\n",
    "# colnames(tbls[['RegionalMarkers']]) <- gsub('RegionalMarkers', '', colnames(tbls[['RegionalMarkers']]))\n",
    "# tbls[['PCmarkers.NicolaTableSelection']] <- merge(tbls[['PCmarkers.NicolaTableSelection']], tbls[['PCmarkers.NicolaTableSelection.topExpressed25']]%>%select(-abs.order), by=c('gene', 'gene.group'), all=T) %>% arrange(abs.order)\n",
    "# tbls[['PCmarkers.NicolaTableSelection.topExpressed25']] <- NULL\n",
    "# colnames(tbls[['PCmarkers.NicolaTableSelection']]) <- gsub('PCmarkers.NicolaTableSelection', '', colnames(tbls[['PCmarkers.NicolaTableSelection']]))\n",
    "# colnames(tbls[['PCtable.topn25']]) <- gsub('PCtable.topn25', '', colnames(tbls[['PCtable.topn25']]))\n",
    "# tbls[['RegionalMarkers']]%>% select(gene, gene.group, expressed)%>%unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8b306c1-01ef-4aa3-b792-d468fa6de951",
   "metadata": {
    "tags": []
   },
   "source": [
    "## All lists of interest"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b5e6e9a3-b5de-48d8-b199-ddaa8ea09836",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Unfiltered lists "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afb9dcbb-22d9-4f15-9837-9c0c2a136bc6",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Unmodified %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "73099699-2e20-4170-bbee-b23cb74e07b3",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists' already exists”\n"
     ]
    }
   ],
   "source": [
    "unfilt_dir <- paste0(original.dir, 'unfiltered_lists/')\n",
    "dir.create(unfilt_dir, recursive = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "7ddc613e-1921-42ec-865b-3f53255ea134",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 53\n"
     ]
    }
   ],
   "source": [
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu, gene.list = genelists.to.plot[[l]], name = l, with_clustering = F,\n",
    "                 save.dir = unfilt_dir)\n",
    "    # stop()\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "d7daf384-0bb4-490b-8104-6e3364f9ec7d",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"#######\"\n",
      "[1] \"Mes prog\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_div_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/Mes prog' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/Mes prog' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 52\n",
      "[1] \"#######\"\n",
      "[1] \"RG early\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_div_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/RG early' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/RG early' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 53\n",
      "[1] \"#######\"\n",
      "[1] \"Neuron\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_div_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/Neuron' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/Neuron' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 50\n",
      "[1] \"#######\"\n",
      "[1] \"RG late\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_div_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/RG late' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/RG late' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 53\n",
      "[1] \"#######\"\n",
      "[1] \"FGF17-like\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_div_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/FGF17-like' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/FGF17-like' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 53\n",
      "[1] \"#######\"\n",
      "[1] \"Neral crest/Mes\"\n",
      "[1] \"#######\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(unfilt_div_dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/Neral crest Mes' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "Warning message in dir.create(save.dir, recursive = T):\n",
      "“'../plots/dotplots_2024-03-06/not_scaled/unfiltered_lists/per_celltype/Neral crest Mes' already exists”\n",
      "NESC progression region markers (Monkey - Selection Fig4D.TableS6.Region_markers_in_vent_NSC_progr)\n",
      "\n",
      "--> Without clustering...\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 (7.69%) repeated genes in FR list\n",
      "0 (0%) repeated genes in MS list\n",
      "5 (33.33%) repeated genes in OC list\n",
      "5 (83.33%) repeated genes in Tem list\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Requiring 3% expression in at least 1 group(s)\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of genes 53 -> 52\n"
     ]
    }
   ],
   "source": [
    "# without clustering\n",
    "for (ct in unique(seu$anno_cluster_fct)){\n",
    "    print('#######')\n",
    "    print(ct)\n",
    "    print('#######')\n",
    "\n",
    "    if (! ct %in% names(ct_seus)){\n",
    "        ct_seus[[ct]] <- ScaleData(subset(seu, anno_cluster_fct == ct), verbose=T)\n",
    "    } \n",
    "    ct_seu <- ct_seus[[ct]]\n",
    "\n",
    "    ct <- gsub(x=ct, pattern='/', replacement=' ' )\n",
    "    unfilt_div_dir <- paste0(original.dir, 'unfiltered_lists/per_celltype/', ct, '/')\n",
    "    dir.create(unfilt_div_dir, recursive = T)\n",
    "\n",
    "    for (l in names(genelists.to.plot)){\n",
    "        message(l)\n",
    "    \n",
    "        ps <- dotplot.save(seu = ct_seu, gene.list = genelists.to.plot[[l]], name = l, with_clustering = F, save.dir = unfilt_div_dir, min.width=2.7)\n",
    "    }\n",
    "    # stop()\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "d179304d-f5da-494f-9bdd-1d5680e3f79b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 980 × 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>avg.exp</th><th scope=col>pct.exp</th><th scope=col>features.plot</th><th scope=col>id</th><th scope=col>avg.exp.scaled</th><th scope=col>feature.groups</th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>celltype</th><th scope=col>Donor</th><th scope=col>repeated</th><th scope=col>exp</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>DLK1</th><td>0.53192249</td><td> 16.279070</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td>-0.16843365</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td>-0.16843365</td></tr>\n",
       "\t<tr><th scope=row>CLYBL</th><td>0.05946106</td><td>  6.976744</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td>-0.13977382</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td>-0.13977382</td></tr>\n",
       "\t<tr><th scope=row>FEZF2</th><td>0.08061623</td><td>  8.139535</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td>-0.25438377</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td>-0.25438377</td></tr>\n",
       "\t<tr><th scope=row>RBP1</th><td>4.68703362</td><td> 94.186047</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td>-0.38419392</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td>-0.38419392</td></tr>\n",
       "\t<tr><th scope=row>FABP7</th><td>1.40068048</td><td> 55.813953</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td> 0.91728912</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td> 0.91728912</td></tr>\n",
       "\t<tr><th scope=row>PTN</th><td>3.66282321</td><td> 86.046512</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td> 0.16276494</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td> 0.16276494</td></tr>\n",
       "\t<tr><th scope=row>NNAT</th><td>1.12007595</td><td> 53.488372</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td> 0.12358644</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td> 0.12358644</td></tr>\n",
       "\t<tr><th scope=row>MEST</th><td>1.09361272</td><td> 54.651163</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_290</td><td>-0.02218101</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_290</td><td>Neral crest/Mes</td><td>D8_Ctrl_290</td><td>Not exclusive</td><td>-0.02218101</td></tr>\n",
       "\t<tr><th scope=row>DLK11</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td>-0.83391610</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td>-0.83391610</td></tr>\n",
       "\t<tr><th scope=row>CLYBL1</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td>-0.94606536</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td>-0.94606536</td></tr>\n",
       "\t<tr><th scope=row>FEZF21</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td>-0.89945076</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td>-0.89945076</td></tr>\n",
       "\t<tr><th scope=row>RBP11</th><td>6.50939878</td><td>100.000000</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td> 1.57747997</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td> 1.57747997</td></tr>\n",
       "\t<tr><th scope=row>FABP71</th><td>1.39256371</td><td> 33.333333</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td> 0.88812940</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td> 0.88812940</td></tr>\n",
       "\t<tr><th scope=row>PTN1</th><td>1.11573859</td><td> 66.666667</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td>-1.59649771</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td>-1.59649771</td></tr>\n",
       "\t<tr><th scope=row>NNAT1</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td>-0.98545725</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td>-0.98545725</td></tr>\n",
       "\t<tr><th scope=row>MEST1</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_Ctrl_311</td><td>-1.32571231</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>Ctrl_311</td><td>Neral crest/Mes</td><td>D8_Ctrl_311</td><td>Not exclusive</td><td>-1.32571231</td></tr>\n",
       "\t<tr><th scope=row>DLK12</th><td>0.81641325</td><td> 33.333333</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 0.18748964</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 0.18748964</td></tr>\n",
       "\t<tr><th scope=row>CLYBL2</th><td>0.16320668</td><td> 11.111111</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 1.26701596</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 1.26701596</td></tr>\n",
       "\t<tr><th scope=row>FEZF22</th><td>0.20028440</td><td> 11.111111</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 0.70316521</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 0.70316521</td></tr>\n",
       "\t<tr><th scope=row>RBP12</th><td>5.38948754</td><td>100.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 0.37195832</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 0.37195832</td></tr>\n",
       "\t<tr><th scope=row>FABP72</th><td>1.19230655</td><td> 66.666667</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 0.16869999</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 0.16869999</td></tr>\n",
       "\t<tr><th scope=row>PTN2</th><td>3.83862711</td><td>100.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 0.28419209</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 0.28419209</td></tr>\n",
       "\t<tr><th scope=row>NNAT2</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td>-0.98545725</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td>-0.98545725</td></tr>\n",
       "\t<tr><th scope=row>MEST2</th><td>1.39129992</td><td> 77.777778</td><td>NA</td><td>Neral crest/Mes_D8_ASD_375 </td><td> 0.33264714</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_375 </td><td>Neral crest/Mes</td><td>D8_ASD_375 </td><td>Not exclusive</td><td> 0.33264714</td></tr>\n",
       "\t<tr><th scope=row>DLK13</th><td>1.95620417</td><td>100.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_384 </td><td> 1.61346968</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_384 </td><td>Neral crest/Mes</td><td>D8_ASD_384 </td><td>Not exclusive</td><td> 1.61346968</td></tr>\n",
       "\t<tr><th scope=row>CLYBL3</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_384 </td><td>-0.94606536</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_384 </td><td>Neral crest/Mes</td><td>D8_ASD_384 </td><td>Not exclusive</td><td>-0.94606536</td></tr>\n",
       "\t<tr><th scope=row>FEZF23</th><td>0.00000000</td><td>  0.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_384 </td><td>-0.89945076</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_384 </td><td>Neral crest/Mes</td><td>D8_ASD_384 </td><td>Not exclusive</td><td>-0.89945076</td></tr>\n",
       "\t<tr><th scope=row>RBP13</th><td>4.33304415</td><td> 66.666667</td><td>NA</td><td>Neral crest/Mes_D8_ASD_384 </td><td>-0.76524373</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_384 </td><td>Neral crest/Mes</td><td>D8_ASD_384 </td><td>Not exclusive</td><td>-0.76524373</td></tr>\n",
       "\t<tr><th scope=row>FABP73</th><td>0.74723953</td><td> 66.666667</td><td>NA</td><td>Neral crest/Mes_D8_ASD_384 </td><td>-1.43021567</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_384 </td><td>Neral crest/Mes</td><td>D8_ASD_384 </td><td>Not exclusive</td><td>-1.43021567</td></tr>\n",
       "\t<tr><th scope=row>PTN3</th><td>5.10553722</td><td>100.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_384 </td><td> 1.15924258</td><td>RPC (PC FGF17)</td><td>Neral crest/Mes</td><td>ASD_384 </td><td>Neral crest/Mes</td><td>D8_ASD_384 </td><td>Not exclusive</td><td> 1.15924258</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>MGST34</th><td>0.78865152</td><td>40.350877</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.81013672</td><td>Antihem (PC SFRP2)</td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.81013672</td></tr>\n",
       "\t<tr><th scope=row>FDPS4</th><td>2.23813533</td><td>69.298246</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.15710735</td><td>Antihem (PC SFRP2)</td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.15710735</td></tr>\n",
       "\t<tr><th scope=row>CSRP24</th><td>2.41223039</td><td>78.947368</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.38241376</td><td>Antihem (PC SFRP2)</td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.38241376</td></tr>\n",
       "\t<tr><th scope=row>NDUFC24</th><td>2.12874383</td><td>78.947368</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.89996486</td><td>Antihem (PC SFRP2)</td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.89996486</td></tr>\n",
       "\t<tr><th scope=row>RHEB4</th><td>1.05587949</td><td>48.245614</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.39524880</td><td>Antihem (PC SFRP2)</td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.39524880</td></tr>\n",
       "\t<tr><th scope=row>WNT44</th><td>0.18960544</td><td> 4.385965</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.47662350</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.47662350</td></tr>\n",
       "\t<tr><th scope=row>IRX34</th><td>0.06361654</td><td> 5.263158</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.81234200</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.81234200</td></tr>\n",
       "\t<tr><th scope=row>FZD104</th><td>0.01312377</td><td> 0.877193</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.39307042</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.39307042</td></tr>\n",
       "\t<tr><th scope=row>TAL24</th><td>0.00000000</td><td> 0.000000</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.44721360</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.44721360</td></tr>\n",
       "\t<tr><th scope=row>APCDD14</th><td>0.06196590</td><td> 3.508772</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 1.45493642</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 1.45493642</td></tr>\n",
       "\t<tr><th scope=row>GRID24</th><td>0.03678118</td><td> 2.631579</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.04023416</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.04023416</td></tr>\n",
       "\t<tr><th scope=row>CRNDE4</th><td>0.46292011</td><td>26.315789</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.57152996</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.57152996</td></tr>\n",
       "\t<tr><th scope=row>OTX14</th><td>0.05160581</td><td> 3.508772</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 1.49713489</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 1.49713489</td></tr>\n",
       "\t<tr><th scope=row>KLHDC8A4</th><td>0.13990294</td><td> 7.894737</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 1.50916496</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 1.50916496</td></tr>\n",
       "\t<tr><th scope=row>IGFBPL14</th><td>0.08393407</td><td> 7.017544</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.24834536</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.24834536</td></tr>\n",
       "\t<tr><th scope=row>LHFPL64</th><td>0.07187527</td><td> 5.263158</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.92905017</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.92905017</td></tr>\n",
       "\t<tr><th scope=row>TIFA4</th><td>0.06455644</td><td> 3.508772</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.46769494</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.46769494</td></tr>\n",
       "\t<tr><th scope=row>POU3F24</th><td>0.27423162</td><td>20.175439</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 1.61177010</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 1.61177010</td></tr>\n",
       "\t<tr><th scope=row>CNTNAP24</th><td>0.42844470</td><td>22.807018</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.71980944</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.71980944</td></tr>\n",
       "\t<tr><th scope=row>TCF7L24</th><td>0.51448694</td><td>33.333333</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.14851779</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.14851779</td></tr>\n",
       "\t<tr><th scope=row>RSRC14</th><td>0.74405803</td><td>40.350877</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.49758605</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.49758605</td></tr>\n",
       "\t<tr><th scope=row>EHBP14</th><td>0.16196183</td><td> 9.649123</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.61449187</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.61449187</td></tr>\n",
       "\t<tr><th scope=row>S100A104</th><td>0.54834629</td><td>31.578947</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.41395272</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.41395272</td></tr>\n",
       "\t<tr><th scope=row>FUOM4</th><td>0.13233939</td><td> 8.771930</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.78506387</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.78506387</td></tr>\n",
       "\t<tr><th scope=row>PPP1R14B4</th><td>3.72956630</td><td>86.842105</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.23559169</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.23559169</td></tr>\n",
       "\t<tr><th scope=row>SLC25A54</th><td>2.50425199</td><td>77.192982</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.33371026</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.33371026</td></tr>\n",
       "\t<tr><th scope=row>SRSF94</th><td>2.17896587</td><td>74.561404</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-1.40011452</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-1.40011452</td></tr>\n",
       "\t<tr><th scope=row>RPL22L14</th><td>2.26044944</td><td>68.421053</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td> 0.05898142</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td> 0.05898142</td></tr>\n",
       "\t<tr><th scope=row>CENPV4</th><td>5.08617288</td><td>92.982456</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.20364216</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.20364216</td></tr>\n",
       "\t<tr><th scope=row>OAZ14</th><td>3.93988554</td><td>86.842105</td><td>NA</td><td>Neral crest/Mes_D8_ASD_494</td><td>-0.12093109</td><td>ZLI (PC TCFL2)    </td><td>Neral crest/Mes</td><td>ASD_494</td><td>Neral crest/Mes</td><td>D8_ASD_494</td><td>Exclusive</td><td>-0.12093109</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 980 × 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & avg.exp & pct.exp & features.plot & id & avg.exp.scaled & feature.groups & V1 & V2 & celltype & Donor & repeated & exp\\\\\n",
       "  & <dbl> & <dbl> & <fct> & <fct> & <dbl> & <fct> & <chr> & <chr> & <fct> & <fct> & <fct> & <dbl>\\\\\n",
       "\\hline\n",
       "\tDLK1 & 0.53192249 &  16.279070 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 & -0.16843365 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive & -0.16843365\\\\\n",
       "\tCLYBL & 0.05946106 &   6.976744 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 & -0.13977382 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive & -0.13977382\\\\\n",
       "\tFEZF2 & 0.08061623 &   8.139535 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 & -0.25438377 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive & -0.25438377\\\\\n",
       "\tRBP1 & 4.68703362 &  94.186047 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 & -0.38419392 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive & -0.38419392\\\\\n",
       "\tFABP7 & 1.40068048 &  55.813953 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 &  0.91728912 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive &  0.91728912\\\\\n",
       "\tPTN & 3.66282321 &  86.046512 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 &  0.16276494 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive &  0.16276494\\\\\n",
       "\tNNAT & 1.12007595 &  53.488372 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 &  0.12358644 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive &  0.12358644\\\\\n",
       "\tMEST & 1.09361272 &  54.651163 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_290 & -0.02218101 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_290 & Neral crest/Mes & D8\\_Ctrl\\_290 & Not exclusive & -0.02218101\\\\\n",
       "\tDLK11 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 & -0.83391610 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive & -0.83391610\\\\\n",
       "\tCLYBL1 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 & -0.94606536 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive & -0.94606536\\\\\n",
       "\tFEZF21 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 & -0.89945076 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive & -0.89945076\\\\\n",
       "\tRBP11 & 6.50939878 & 100.000000 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 &  1.57747997 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive &  1.57747997\\\\\n",
       "\tFABP71 & 1.39256371 &  33.333333 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 &  0.88812940 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive &  0.88812940\\\\\n",
       "\tPTN1 & 1.11573859 &  66.666667 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 & -1.59649771 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive & -1.59649771\\\\\n",
       "\tNNAT1 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 & -0.98545725 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive & -0.98545725\\\\\n",
       "\tMEST1 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_Ctrl\\_311 & -1.32571231 & RPC (PC FGF17) & Neral crest/Mes & Ctrl\\_311 & Neral crest/Mes & D8\\_Ctrl\\_311 & Not exclusive & -1.32571231\\\\\n",
       "\tDLK12 & 0.81641325 &  33.333333 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  0.18748964 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  0.18748964\\\\\n",
       "\tCLYBL2 & 0.16320668 &  11.111111 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  1.26701596 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  1.26701596\\\\\n",
       "\tFEZF22 & 0.20028440 &  11.111111 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  0.70316521 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  0.70316521\\\\\n",
       "\tRBP12 & 5.38948754 & 100.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  0.37195832 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  0.37195832\\\\\n",
       "\tFABP72 & 1.19230655 &  66.666667 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  0.16869999 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  0.16869999\\\\\n",
       "\tPTN2 & 3.83862711 & 100.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  0.28419209 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  0.28419209\\\\\n",
       "\tNNAT2 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  & -0.98545725 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive & -0.98545725\\\\\n",
       "\tMEST2 & 1.39129992 &  77.777778 & NA & Neral crest/Mes\\_D8\\_ASD\\_375  &  0.33264714 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_375  & Neral crest/Mes & D8\\_ASD\\_375  & Not exclusive &  0.33264714\\\\\n",
       "\tDLK13 & 1.95620417 & 100.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_384  &  1.61346968 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_384  & Neral crest/Mes & D8\\_ASD\\_384  & Not exclusive &  1.61346968\\\\\n",
       "\tCLYBL3 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_384  & -0.94606536 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_384  & Neral crest/Mes & D8\\_ASD\\_384  & Not exclusive & -0.94606536\\\\\n",
       "\tFEZF23 & 0.00000000 &   0.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_384  & -0.89945076 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_384  & Neral crest/Mes & D8\\_ASD\\_384  & Not exclusive & -0.89945076\\\\\n",
       "\tRBP13 & 4.33304415 &  66.666667 & NA & Neral crest/Mes\\_D8\\_ASD\\_384  & -0.76524373 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_384  & Neral crest/Mes & D8\\_ASD\\_384  & Not exclusive & -0.76524373\\\\\n",
       "\tFABP73 & 0.74723953 &  66.666667 & NA & Neral crest/Mes\\_D8\\_ASD\\_384  & -1.43021567 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_384  & Neral crest/Mes & D8\\_ASD\\_384  & Not exclusive & -1.43021567\\\\\n",
       "\tPTN3 & 5.10553722 & 100.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_384  &  1.15924258 & RPC (PC FGF17) & Neral crest/Mes & ASD\\_384  & Neral crest/Mes & D8\\_ASD\\_384  & Not exclusive &  1.15924258\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tMGST34 & 0.78865152 & 40.350877 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.81013672 & Antihem (PC SFRP2) & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.81013672\\\\\n",
       "\tFDPS4 & 2.23813533 & 69.298246 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.15710735 & Antihem (PC SFRP2) & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.15710735\\\\\n",
       "\tCSRP24 & 2.41223039 & 78.947368 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.38241376 & Antihem (PC SFRP2) & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.38241376\\\\\n",
       "\tNDUFC24 & 2.12874383 & 78.947368 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.89996486 & Antihem (PC SFRP2) & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.89996486\\\\\n",
       "\tRHEB4 & 1.05587949 & 48.245614 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.39524880 & Antihem (PC SFRP2) & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.39524880\\\\\n",
       "\tWNT44 & 0.18960544 &  4.385965 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.47662350 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.47662350\\\\\n",
       "\tIRX34 & 0.06361654 &  5.263158 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.81234200 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.81234200\\\\\n",
       "\tFZD104 & 0.01312377 &  0.877193 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.39307042 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.39307042\\\\\n",
       "\tTAL24 & 0.00000000 &  0.000000 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.44721360 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.44721360\\\\\n",
       "\tAPCDD14 & 0.06196590 &  3.508772 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  1.45493642 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  1.45493642\\\\\n",
       "\tGRID24 & 0.03678118 &  2.631579 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.04023416 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.04023416\\\\\n",
       "\tCRNDE4 & 0.46292011 & 26.315789 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.57152996 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.57152996\\\\\n",
       "\tOTX14 & 0.05160581 &  3.508772 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  1.49713489 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  1.49713489\\\\\n",
       "\tKLHDC8A4 & 0.13990294 &  7.894737 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  1.50916496 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  1.50916496\\\\\n",
       "\tIGFBPL14 & 0.08393407 &  7.017544 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.24834536 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.24834536\\\\\n",
       "\tLHFPL64 & 0.07187527 &  5.263158 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.92905017 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.92905017\\\\\n",
       "\tTIFA4 & 0.06455644 &  3.508772 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.46769494 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.46769494\\\\\n",
       "\tPOU3F24 & 0.27423162 & 20.175439 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  1.61177010 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  1.61177010\\\\\n",
       "\tCNTNAP24 & 0.42844470 & 22.807018 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.71980944 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.71980944\\\\\n",
       "\tTCF7L24 & 0.51448694 & 33.333333 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.14851779 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.14851779\\\\\n",
       "\tRSRC14 & 0.74405803 & 40.350877 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.49758605 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.49758605\\\\\n",
       "\tEHBP14 & 0.16196183 &  9.649123 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.61449187 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.61449187\\\\\n",
       "\tS100A104 & 0.54834629 & 31.578947 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.41395272 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.41395272\\\\\n",
       "\tFUOM4 & 0.13233939 &  8.771930 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.78506387 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.78506387\\\\\n",
       "\tPPP1R14B4 & 3.72956630 & 86.842105 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.23559169 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.23559169\\\\\n",
       "\tSLC25A54 & 2.50425199 & 77.192982 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.33371026 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.33371026\\\\\n",
       "\tSRSF94 & 2.17896587 & 74.561404 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -1.40011452 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -1.40011452\\\\\n",
       "\tRPL22L14 & 2.26044944 & 68.421053 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 &  0.05898142 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive &  0.05898142\\\\\n",
       "\tCENPV4 & 5.08617288 & 92.982456 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.20364216 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.20364216\\\\\n",
       "\tOAZ14 & 3.93988554 & 86.842105 & NA & Neral crest/Mes\\_D8\\_ASD\\_494 & -0.12093109 & ZLI (PC TCFL2)     & Neral crest/Mes & ASD\\_494 & Neral crest/Mes & D8\\_ASD\\_494 & Exclusive & -0.12093109\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 980 × 12\n",
       "\n",
       "| <!--/--> | avg.exp &lt;dbl&gt; | pct.exp &lt;dbl&gt; | features.plot &lt;fct&gt; | id &lt;fct&gt; | avg.exp.scaled &lt;dbl&gt; | feature.groups &lt;fct&gt; | V1 &lt;chr&gt; | V2 &lt;chr&gt; | celltype &lt;fct&gt; | Donor &lt;fct&gt; | repeated &lt;fct&gt; | exp &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| DLK1 | 0.53192249 |  16.279070 | NA | Neral crest/Mes_D8_Ctrl_290 | -0.16843365 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive | -0.16843365 |\n",
       "| CLYBL | 0.05946106 |   6.976744 | NA | Neral crest/Mes_D8_Ctrl_290 | -0.13977382 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive | -0.13977382 |\n",
       "| FEZF2 | 0.08061623 |   8.139535 | NA | Neral crest/Mes_D8_Ctrl_290 | -0.25438377 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive | -0.25438377 |\n",
       "| RBP1 | 4.68703362 |  94.186047 | NA | Neral crest/Mes_D8_Ctrl_290 | -0.38419392 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive | -0.38419392 |\n",
       "| FABP7 | 1.40068048 |  55.813953 | NA | Neral crest/Mes_D8_Ctrl_290 |  0.91728912 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive |  0.91728912 |\n",
       "| PTN | 3.66282321 |  86.046512 | NA | Neral crest/Mes_D8_Ctrl_290 |  0.16276494 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive |  0.16276494 |\n",
       "| NNAT | 1.12007595 |  53.488372 | NA | Neral crest/Mes_D8_Ctrl_290 |  0.12358644 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive |  0.12358644 |\n",
       "| MEST | 1.09361272 |  54.651163 | NA | Neral crest/Mes_D8_Ctrl_290 | -0.02218101 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_290 | Neral crest/Mes | D8_Ctrl_290 | Not exclusive | -0.02218101 |\n",
       "| DLK11 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_Ctrl_311 | -0.83391610 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive | -0.83391610 |\n",
       "| CLYBL1 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_Ctrl_311 | -0.94606536 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive | -0.94606536 |\n",
       "| FEZF21 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_Ctrl_311 | -0.89945076 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive | -0.89945076 |\n",
       "| RBP11 | 6.50939878 | 100.000000 | NA | Neral crest/Mes_D8_Ctrl_311 |  1.57747997 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive |  1.57747997 |\n",
       "| FABP71 | 1.39256371 |  33.333333 | NA | Neral crest/Mes_D8_Ctrl_311 |  0.88812940 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive |  0.88812940 |\n",
       "| PTN1 | 1.11573859 |  66.666667 | NA | Neral crest/Mes_D8_Ctrl_311 | -1.59649771 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive | -1.59649771 |\n",
       "| NNAT1 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_Ctrl_311 | -0.98545725 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive | -0.98545725 |\n",
       "| MEST1 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_Ctrl_311 | -1.32571231 | RPC (PC FGF17) | Neral crest/Mes | Ctrl_311 | Neral crest/Mes | D8_Ctrl_311 | Not exclusive | -1.32571231 |\n",
       "| DLK12 | 0.81641325 |  33.333333 | NA | Neral crest/Mes_D8_ASD_375  |  0.18748964 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  0.18748964 |\n",
       "| CLYBL2 | 0.16320668 |  11.111111 | NA | Neral crest/Mes_D8_ASD_375  |  1.26701596 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  1.26701596 |\n",
       "| FEZF22 | 0.20028440 |  11.111111 | NA | Neral crest/Mes_D8_ASD_375  |  0.70316521 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  0.70316521 |\n",
       "| RBP12 | 5.38948754 | 100.000000 | NA | Neral crest/Mes_D8_ASD_375  |  0.37195832 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  0.37195832 |\n",
       "| FABP72 | 1.19230655 |  66.666667 | NA | Neral crest/Mes_D8_ASD_375  |  0.16869999 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  0.16869999 |\n",
       "| PTN2 | 3.83862711 | 100.000000 | NA | Neral crest/Mes_D8_ASD_375  |  0.28419209 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  0.28419209 |\n",
       "| NNAT2 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_ASD_375  | -0.98545725 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive | -0.98545725 |\n",
       "| MEST2 | 1.39129992 |  77.777778 | NA | Neral crest/Mes_D8_ASD_375  |  0.33264714 | RPC (PC FGF17) | Neral crest/Mes | ASD_375  | Neral crest/Mes | D8_ASD_375  | Not exclusive |  0.33264714 |\n",
       "| DLK13 | 1.95620417 | 100.000000 | NA | Neral crest/Mes_D8_ASD_384  |  1.61346968 | RPC (PC FGF17) | Neral crest/Mes | ASD_384  | Neral crest/Mes | D8_ASD_384  | Not exclusive |  1.61346968 |\n",
       "| CLYBL3 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_ASD_384  | -0.94606536 | RPC (PC FGF17) | Neral crest/Mes | ASD_384  | Neral crest/Mes | D8_ASD_384  | Not exclusive | -0.94606536 |\n",
       "| FEZF23 | 0.00000000 |   0.000000 | NA | Neral crest/Mes_D8_ASD_384  | -0.89945076 | RPC (PC FGF17) | Neral crest/Mes | ASD_384  | Neral crest/Mes | D8_ASD_384  | Not exclusive | -0.89945076 |\n",
       "| RBP13 | 4.33304415 |  66.666667 | NA | Neral crest/Mes_D8_ASD_384  | -0.76524373 | RPC (PC FGF17) | Neral crest/Mes | ASD_384  | Neral crest/Mes | D8_ASD_384  | Not exclusive | -0.76524373 |\n",
       "| FABP73 | 0.74723953 |  66.666667 | NA | Neral crest/Mes_D8_ASD_384  | -1.43021567 | RPC (PC FGF17) | Neral crest/Mes | ASD_384  | Neral crest/Mes | D8_ASD_384  | Not exclusive | -1.43021567 |\n",
       "| PTN3 | 5.10553722 | 100.000000 | NA | Neral crest/Mes_D8_ASD_384  |  1.15924258 | RPC (PC FGF17) | Neral crest/Mes | ASD_384  | Neral crest/Mes | D8_ASD_384  | Not exclusive |  1.15924258 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| MGST34 | 0.78865152 | 40.350877 | NA | Neral crest/Mes_D8_ASD_494 | -0.81013672 | Antihem (PC SFRP2) | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.81013672 |\n",
       "| FDPS4 | 2.23813533 | 69.298246 | NA | Neral crest/Mes_D8_ASD_494 | -0.15710735 | Antihem (PC SFRP2) | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.15710735 |\n",
       "| CSRP24 | 2.41223039 | 78.947368 | NA | Neral crest/Mes_D8_ASD_494 | -0.38241376 | Antihem (PC SFRP2) | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.38241376 |\n",
       "| NDUFC24 | 2.12874383 | 78.947368 | NA | Neral crest/Mes_D8_ASD_494 | -0.89996486 | Antihem (PC SFRP2) | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.89996486 |\n",
       "| RHEB4 | 1.05587949 | 48.245614 | NA | Neral crest/Mes_D8_ASD_494 | -0.39524880 | Antihem (PC SFRP2) | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.39524880 |\n",
       "| WNT44 | 0.18960544 |  4.385965 | NA | Neral crest/Mes_D8_ASD_494 |  0.47662350 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.47662350 |\n",
       "| IRX34 | 0.06361654 |  5.263158 | NA | Neral crest/Mes_D8_ASD_494 |  0.81234200 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.81234200 |\n",
       "| FZD104 | 0.01312377 |  0.877193 | NA | Neral crest/Mes_D8_ASD_494 | -0.39307042 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.39307042 |\n",
       "| TAL24 | 0.00000000 |  0.000000 | NA | Neral crest/Mes_D8_ASD_494 | -0.44721360 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.44721360 |\n",
       "| APCDD14 | 0.06196590 |  3.508772 | NA | Neral crest/Mes_D8_ASD_494 |  1.45493642 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  1.45493642 |\n",
       "| GRID24 | 0.03678118 |  2.631579 | NA | Neral crest/Mes_D8_ASD_494 |  0.04023416 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.04023416 |\n",
       "| CRNDE4 | 0.46292011 | 26.315789 | NA | Neral crest/Mes_D8_ASD_494 | -0.57152996 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.57152996 |\n",
       "| OTX14 | 0.05160581 |  3.508772 | NA | Neral crest/Mes_D8_ASD_494 |  1.49713489 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  1.49713489 |\n",
       "| KLHDC8A4 | 0.13990294 |  7.894737 | NA | Neral crest/Mes_D8_ASD_494 |  1.50916496 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  1.50916496 |\n",
       "| IGFBPL14 | 0.08393407 |  7.017544 | NA | Neral crest/Mes_D8_ASD_494 | -0.24834536 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.24834536 |\n",
       "| LHFPL64 | 0.07187527 |  5.263158 | NA | Neral crest/Mes_D8_ASD_494 |  0.92905017 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.92905017 |\n",
       "| TIFA4 | 0.06455644 |  3.508772 | NA | Neral crest/Mes_D8_ASD_494 | -0.46769494 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.46769494 |\n",
       "| POU3F24 | 0.27423162 | 20.175439 | NA | Neral crest/Mes_D8_ASD_494 |  1.61177010 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  1.61177010 |\n",
       "| CNTNAP24 | 0.42844470 | 22.807018 | NA | Neral crest/Mes_D8_ASD_494 |  0.71980944 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.71980944 |\n",
       "| TCF7L24 | 0.51448694 | 33.333333 | NA | Neral crest/Mes_D8_ASD_494 |  0.14851779 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.14851779 |\n",
       "| RSRC14 | 0.74405803 | 40.350877 | NA | Neral crest/Mes_D8_ASD_494 | -0.49758605 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.49758605 |\n",
       "| EHBP14 | 0.16196183 |  9.649123 | NA | Neral crest/Mes_D8_ASD_494 | -0.61449187 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.61449187 |\n",
       "| S100A104 | 0.54834629 | 31.578947 | NA | Neral crest/Mes_D8_ASD_494 |  0.41395272 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.41395272 |\n",
       "| FUOM4 | 0.13233939 |  8.771930 | NA | Neral crest/Mes_D8_ASD_494 |  0.78506387 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.78506387 |\n",
       "| PPP1R14B4 | 3.72956630 | 86.842105 | NA | Neral crest/Mes_D8_ASD_494 |  0.23559169 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.23559169 |\n",
       "| SLC25A54 | 2.50425199 | 77.192982 | NA | Neral crest/Mes_D8_ASD_494 | -0.33371026 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.33371026 |\n",
       "| SRSF94 | 2.17896587 | 74.561404 | NA | Neral crest/Mes_D8_ASD_494 | -1.40011452 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -1.40011452 |\n",
       "| RPL22L14 | 2.26044944 | 68.421053 | NA | Neral crest/Mes_D8_ASD_494 |  0.05898142 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive |  0.05898142 |\n",
       "| CENPV4 | 5.08617288 | 92.982456 | NA | Neral crest/Mes_D8_ASD_494 | -0.20364216 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.20364216 |\n",
       "| OAZ14 | 3.93988554 | 86.842105 | NA | Neral crest/Mes_D8_ASD_494 | -0.12093109 | ZLI (PC TCFL2)     | Neral crest/Mes | ASD_494 | Neral crest/Mes | D8_ASD_494 | Exclusive | -0.12093109 |\n",
       "\n"
      ],
      "text/plain": [
       "          avg.exp    pct.exp    features.plot id                         \n",
       "DLK1      0.53192249  16.279070 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "CLYBL     0.05946106   6.976744 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "FEZF2     0.08061623   8.139535 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "RBP1      4.68703362  94.186047 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "FABP7     1.40068048  55.813953 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "PTN       3.66282321  86.046512 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "NNAT      1.12007595  53.488372 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "MEST      1.09361272  54.651163 NA            Neral crest/Mes_D8_Ctrl_290\n",
       "DLK11     0.00000000   0.000000 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "CLYBL1    0.00000000   0.000000 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "FEZF21    0.00000000   0.000000 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "RBP11     6.50939878 100.000000 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "FABP71    1.39256371  33.333333 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "PTN1      1.11573859  66.666667 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "NNAT1     0.00000000   0.000000 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "MEST1     0.00000000   0.000000 NA            Neral crest/Mes_D8_Ctrl_311\n",
       "DLK12     0.81641325  33.333333 NA            Neral crest/Mes_D8_ASD_375 \n",
       "CLYBL2    0.16320668  11.111111 NA            Neral crest/Mes_D8_ASD_375 \n",
       "FEZF22    0.20028440  11.111111 NA            Neral crest/Mes_D8_ASD_375 \n",
       "RBP12     5.38948754 100.000000 NA            Neral crest/Mes_D8_ASD_375 \n",
       "FABP72    1.19230655  66.666667 NA            Neral crest/Mes_D8_ASD_375 \n",
       "PTN2      3.83862711 100.000000 NA            Neral crest/Mes_D8_ASD_375 \n",
       "NNAT2     0.00000000   0.000000 NA            Neral crest/Mes_D8_ASD_375 \n",
       "MEST2     1.39129992  77.777778 NA            Neral crest/Mes_D8_ASD_375 \n",
       "DLK13     1.95620417 100.000000 NA            Neral crest/Mes_D8_ASD_384 \n",
       "CLYBL3    0.00000000   0.000000 NA            Neral crest/Mes_D8_ASD_384 \n",
       "FEZF23    0.00000000   0.000000 NA            Neral crest/Mes_D8_ASD_384 \n",
       "RBP13     4.33304415  66.666667 NA            Neral crest/Mes_D8_ASD_384 \n",
       "FABP73    0.74723953  66.666667 NA            Neral crest/Mes_D8_ASD_384 \n",
       "PTN3      5.10553722 100.000000 NA            Neral crest/Mes_D8_ASD_384 \n",
       "⋮         ⋮          ⋮          ⋮             ⋮                          \n",
       "MGST34    0.78865152 40.350877  NA            Neral crest/Mes_D8_ASD_494 \n",
       "FDPS4     2.23813533 69.298246  NA            Neral crest/Mes_D8_ASD_494 \n",
       "CSRP24    2.41223039 78.947368  NA            Neral crest/Mes_D8_ASD_494 \n",
       "NDUFC24   2.12874383 78.947368  NA            Neral crest/Mes_D8_ASD_494 \n",
       "RHEB4     1.05587949 48.245614  NA            Neral crest/Mes_D8_ASD_494 \n",
       "WNT44     0.18960544  4.385965  NA            Neral crest/Mes_D8_ASD_494 \n",
       "IRX34     0.06361654  5.263158  NA            Neral crest/Mes_D8_ASD_494 \n",
       "FZD104    0.01312377  0.877193  NA            Neral crest/Mes_D8_ASD_494 \n",
       "TAL24     0.00000000  0.000000  NA            Neral crest/Mes_D8_ASD_494 \n",
       "APCDD14   0.06196590  3.508772  NA            Neral crest/Mes_D8_ASD_494 \n",
       "GRID24    0.03678118  2.631579  NA            Neral crest/Mes_D8_ASD_494 \n",
       "CRNDE4    0.46292011 26.315789  NA            Neral crest/Mes_D8_ASD_494 \n",
       "OTX14     0.05160581  3.508772  NA            Neral crest/Mes_D8_ASD_494 \n",
       "KLHDC8A4  0.13990294  7.894737  NA            Neral crest/Mes_D8_ASD_494 \n",
       "IGFBPL14  0.08393407  7.017544  NA            Neral crest/Mes_D8_ASD_494 \n",
       "LHFPL64   0.07187527  5.263158  NA            Neral crest/Mes_D8_ASD_494 \n",
       "TIFA4     0.06455644  3.508772  NA            Neral crest/Mes_D8_ASD_494 \n",
       "POU3F24   0.27423162 20.175439  NA            Neral crest/Mes_D8_ASD_494 \n",
       "CNTNAP24  0.42844470 22.807018  NA            Neral crest/Mes_D8_ASD_494 \n",
       "TCF7L24   0.51448694 33.333333  NA            Neral crest/Mes_D8_ASD_494 \n",
       "RSRC14    0.74405803 40.350877  NA            Neral crest/Mes_D8_ASD_494 \n",
       "EHBP14    0.16196183  9.649123  NA            Neral crest/Mes_D8_ASD_494 \n",
       "S100A104  0.54834629 31.578947  NA            Neral crest/Mes_D8_ASD_494 \n",
       "FUOM4     0.13233939  8.771930  NA            Neral crest/Mes_D8_ASD_494 \n",
       "PPP1R14B4 3.72956630 86.842105  NA            Neral crest/Mes_D8_ASD_494 \n",
       "SLC25A54  2.50425199 77.192982  NA            Neral crest/Mes_D8_ASD_494 \n",
       "SRSF94    2.17896587 74.561404  NA            Neral crest/Mes_D8_ASD_494 \n",
       "RPL22L14  2.26044944 68.421053  NA            Neral crest/Mes_D8_ASD_494 \n",
       "CENPV4    5.08617288 92.982456  NA            Neral crest/Mes_D8_ASD_494 \n",
       "OAZ14     3.93988554 86.842105  NA            Neral crest/Mes_D8_ASD_494 \n",
       "          avg.exp.scaled feature.groups     V1              V2      \n",
       "DLK1      -0.16843365    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "CLYBL     -0.13977382    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "FEZF2     -0.25438377    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "RBP1      -0.38419392    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "FABP7      0.91728912    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "PTN        0.16276494    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "NNAT       0.12358644    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "MEST      -0.02218101    RPC (PC FGF17)     Neral crest/Mes Ctrl_290\n",
       "DLK11     -0.83391610    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "CLYBL1    -0.94606536    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "FEZF21    -0.89945076    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "RBP11      1.57747997    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "FABP71     0.88812940    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "PTN1      -1.59649771    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "NNAT1     -0.98545725    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "MEST1     -1.32571231    RPC (PC FGF17)     Neral crest/Mes Ctrl_311\n",
       "DLK12      0.18748964    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "CLYBL2     1.26701596    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "FEZF22     0.70316521    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "RBP12      0.37195832    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "FABP72     0.16869999    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "PTN2       0.28419209    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "NNAT2     -0.98545725    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "MEST2      0.33264714    RPC (PC FGF17)     Neral crest/Mes ASD_375 \n",
       "DLK13      1.61346968    RPC (PC FGF17)     Neral crest/Mes ASD_384 \n",
       "CLYBL3    -0.94606536    RPC (PC FGF17)     Neral crest/Mes ASD_384 \n",
       "FEZF23    -0.89945076    RPC (PC FGF17)     Neral crest/Mes ASD_384 \n",
       "RBP13     -0.76524373    RPC (PC FGF17)     Neral crest/Mes ASD_384 \n",
       "FABP73    -1.43021567    RPC (PC FGF17)     Neral crest/Mes ASD_384 \n",
       "PTN3       1.15924258    RPC (PC FGF17)     Neral crest/Mes ASD_384 \n",
       "⋮         ⋮              ⋮                  ⋮               ⋮       \n",
       "MGST34    -0.81013672    Antihem (PC SFRP2) Neral crest/Mes ASD_494 \n",
       "FDPS4     -0.15710735    Antihem (PC SFRP2) Neral crest/Mes ASD_494 \n",
       "CSRP24    -0.38241376    Antihem (PC SFRP2) Neral crest/Mes ASD_494 \n",
       "NDUFC24   -0.89996486    Antihem (PC SFRP2) Neral crest/Mes ASD_494 \n",
       "RHEB4     -0.39524880    Antihem (PC SFRP2) Neral crest/Mes ASD_494 \n",
       "WNT44      0.47662350    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "IRX34      0.81234200    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "FZD104    -0.39307042    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "TAL24     -0.44721360    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "APCDD14    1.45493642    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "GRID24     0.04023416    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "CRNDE4    -0.57152996    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "OTX14      1.49713489    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "KLHDC8A4   1.50916496    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "IGFBPL14  -0.24834536    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "LHFPL64    0.92905017    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "TIFA4     -0.46769494    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "POU3F24    1.61177010    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "CNTNAP24   0.71980944    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "TCF7L24    0.14851779    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "RSRC14    -0.49758605    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "EHBP14    -0.61449187    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "S100A104   0.41395272    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "FUOM4      0.78506387    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "PPP1R14B4  0.23559169    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "SLC25A54  -0.33371026    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "SRSF94    -1.40011452    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "RPL22L14   0.05898142    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "CENPV4    -0.20364216    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "OAZ14     -0.12093109    ZLI (PC TCFL2)     Neral crest/Mes ASD_494 \n",
       "          celltype        Donor       repeated      exp        \n",
       "DLK1      Neral crest/Mes D8_Ctrl_290 Not exclusive -0.16843365\n",
       "CLYBL     Neral crest/Mes D8_Ctrl_290 Not exclusive -0.13977382\n",
       "FEZF2     Neral crest/Mes D8_Ctrl_290 Not exclusive -0.25438377\n",
       "RBP1      Neral crest/Mes D8_Ctrl_290 Not exclusive -0.38419392\n",
       "FABP7     Neral crest/Mes D8_Ctrl_290 Not exclusive  0.91728912\n",
       "PTN       Neral crest/Mes D8_Ctrl_290 Not exclusive  0.16276494\n",
       "NNAT      Neral crest/Mes D8_Ctrl_290 Not exclusive  0.12358644\n",
       "MEST      Neral crest/Mes D8_Ctrl_290 Not exclusive -0.02218101\n",
       "DLK11     Neral crest/Mes D8_Ctrl_311 Not exclusive -0.83391610\n",
       "CLYBL1    Neral crest/Mes D8_Ctrl_311 Not exclusive -0.94606536\n",
       "FEZF21    Neral crest/Mes D8_Ctrl_311 Not exclusive -0.89945076\n",
       "RBP11     Neral crest/Mes D8_Ctrl_311 Not exclusive  1.57747997\n",
       "FABP71    Neral crest/Mes D8_Ctrl_311 Not exclusive  0.88812940\n",
       "PTN1      Neral crest/Mes D8_Ctrl_311 Not exclusive -1.59649771\n",
       "NNAT1     Neral crest/Mes D8_Ctrl_311 Not exclusive -0.98545725\n",
       "MEST1     Neral crest/Mes D8_Ctrl_311 Not exclusive -1.32571231\n",
       "DLK12     Neral crest/Mes D8_ASD_375  Not exclusive  0.18748964\n",
       "CLYBL2    Neral crest/Mes D8_ASD_375  Not exclusive  1.26701596\n",
       "FEZF22    Neral crest/Mes D8_ASD_375  Not exclusive  0.70316521\n",
       "RBP12     Neral crest/Mes D8_ASD_375  Not exclusive  0.37195832\n",
       "FABP72    Neral crest/Mes D8_ASD_375  Not exclusive  0.16869999\n",
       "PTN2      Neral crest/Mes D8_ASD_375  Not exclusive  0.28419209\n",
       "NNAT2     Neral crest/Mes D8_ASD_375  Not exclusive -0.98545725\n",
       "MEST2     Neral crest/Mes D8_ASD_375  Not exclusive  0.33264714\n",
       "DLK13     Neral crest/Mes D8_ASD_384  Not exclusive  1.61346968\n",
       "CLYBL3    Neral crest/Mes D8_ASD_384  Not exclusive -0.94606536\n",
       "FEZF23    Neral crest/Mes D8_ASD_384  Not exclusive -0.89945076\n",
       "RBP13     Neral crest/Mes D8_ASD_384  Not exclusive -0.76524373\n",
       "FABP73    Neral crest/Mes D8_ASD_384  Not exclusive -1.43021567\n",
       "PTN3      Neral crest/Mes D8_ASD_384  Not exclusive  1.15924258\n",
       "⋮         ⋮               ⋮           ⋮             ⋮          \n",
       "MGST34    Neral crest/Mes D8_ASD_494  Exclusive     -0.81013672\n",
       "FDPS4     Neral crest/Mes D8_ASD_494  Exclusive     -0.15710735\n",
       "CSRP24    Neral crest/Mes D8_ASD_494  Exclusive     -0.38241376\n",
       "NDUFC24   Neral crest/Mes D8_ASD_494  Exclusive     -0.89996486\n",
       "RHEB4     Neral crest/Mes D8_ASD_494  Exclusive     -0.39524880\n",
       "WNT44     Neral crest/Mes D8_ASD_494  Exclusive      0.47662350\n",
       "IRX34     Neral crest/Mes D8_ASD_494  Exclusive      0.81234200\n",
       "FZD104    Neral crest/Mes D8_ASD_494  Exclusive     -0.39307042\n",
       "TAL24     Neral crest/Mes D8_ASD_494  Exclusive     -0.44721360\n",
       "APCDD14   Neral crest/Mes D8_ASD_494  Exclusive      1.45493642\n",
       "GRID24    Neral crest/Mes D8_ASD_494  Exclusive      0.04023416\n",
       "CRNDE4    Neral crest/Mes D8_ASD_494  Exclusive     -0.57152996\n",
       "OTX14     Neral crest/Mes D8_ASD_494  Exclusive      1.49713489\n",
       "KLHDC8A4  Neral crest/Mes D8_ASD_494  Exclusive      1.50916496\n",
       "IGFBPL14  Neral crest/Mes D8_ASD_494  Exclusive     -0.24834536\n",
       "LHFPL64   Neral crest/Mes D8_ASD_494  Exclusive      0.92905017\n",
       "TIFA4     Neral crest/Mes D8_ASD_494  Exclusive     -0.46769494\n",
       "POU3F24   Neral crest/Mes D8_ASD_494  Exclusive      1.61177010\n",
       "CNTNAP24  Neral crest/Mes D8_ASD_494  Exclusive      0.71980944\n",
       "TCF7L24   Neral crest/Mes D8_ASD_494  Exclusive      0.14851779\n",
       "RSRC14    Neral crest/Mes D8_ASD_494  Exclusive     -0.49758605\n",
       "EHBP14    Neral crest/Mes D8_ASD_494  Exclusive     -0.61449187\n",
       "S100A104  Neral crest/Mes D8_ASD_494  Exclusive      0.41395272\n",
       "FUOM4     Neral crest/Mes D8_ASD_494  Exclusive      0.78506387\n",
       "PPP1R14B4 Neral crest/Mes D8_ASD_494  Exclusive      0.23559169\n",
       "SLC25A54  Neral crest/Mes D8_ASD_494  Exclusive     -0.33371026\n",
       "SRSF94    Neral crest/Mes D8_ASD_494  Exclusive     -1.40011452\n",
       "RPL22L14  Neral crest/Mes D8_ASD_494  Exclusive      0.05898142\n",
       "CENPV4    Neral crest/Mes D8_ASD_494  Exclusive     -0.20364216\n",
       "OAZ14     Neral crest/Mes D8_ASD_494  Exclusive     -0.12093109"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ps$`not clust`$data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "538d6e96-dc17-4ffa-80f8-eeee5f6f6bf7",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "# Other"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ddab8b9-be2c-42d0-89c2-3817960194df",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Selected marker genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3ebfb34-b82e-4674-961a-53d65d70b34e",
   "metadata": {},
   "source": [
    "### Per donor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90aef16c-ea94-463d-8391-5a6277657aef",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "markers.donors <- data.table::fread('../results/selected_markers/markers_per_donor.nohighmito.with_filters_and_crossed.csv', data.table=F, drop = 1)\n",
    "\n",
    "# markers.donor.lists <- list()\n",
    "for (x in c('filter.padj','filter.padj.top100','filter.pval.top100')){\n",
    "    \n",
    "    filt.dat <- markers.donors[markers.donors[,x]&!markers.donors$in.sex_chr, ]\n",
    "    # markers.donor.lists[[x]] <- lapply(split(filt.dat, filt.dat[,'anno_cluster_fct']),\n",
    "    #                                    function(y){\n",
    "    #                                        split(y, y[, 'overexpressed.in']) %>% lapply(function(z){\n",
    "    #                                            split(z$gene, z[, paste0(x, '.crossed')]) %>% lapply(unique)\n",
    "    #                                         })\n",
    "    #                                    })\n",
    "    oe.marker.list <- split(filt.dat, filt.dat$overexpressed.in)\n",
    "    \n",
    "    \n",
    "    for (oe in names(oe.marker.list)){\n",
    "        ct.inters.list <- split(oe.marker.list[[oe]], oe.marker.list[[oe]]$anno_cluster_fct) \n",
    "        \n",
    "        for (ct in names(ct.inters.list)){\n",
    "            dinters.list <- lapply(split(ct.inters.list[[ct]]$gene, ct.inters.list[[ct]][, paste0(x, '.crossed')]), unique)\n",
    "            ct <- gsub(pattern = '/', replacement = '_', x = ct, fixed = T)\n",
    "            dotplot.save(with_clustering = F, gene.list = dinters.list, seu = seu,save.dir = paste0(mdonor_dir, oe, '/'),\n",
    "                         name=glue::glue('{x}.{ct}.Differentially expressed genes (ASD vs Ctrl) per donor'))\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1faad56e-7be4-43f8-9dfb-f2665558a9e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(ct.inters.list)%>%\n",
    "gsub(pattern='ASD.', fixed=T, replacement = 'Higher in ASD\\n') %>%\n",
    "gsub(pattern='Ctrl.', fixed=T, replacement = 'Higher in Ctrl\\n')%>%\n",
    "# gsub(pattern='&', fixed=T, replacement = ' &\\n')%>%\n",
    "gsub(pattern='D8_', fixed=T, replacement = '') -> names(ct.inters.list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f8f349a-1f9b-4409-92e4-6273557b2b4f",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "fig(30,15)\n",
    "dotplot.save(gene.list = ct.inters.list, seu = seu,save.dir = mdonor_dir, name='Differentially expressed genes (ASD vs Ctrl) per donor')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66aef3cc-e8b7-4572-9d90-b26bc43f862c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "mdonor_dir <- paste0(today_dir, 'markers_per_donor/')\n",
    "dir.create(mdonor_dir, recursive = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9da7d40-15ba-4de7-9b86-90458a4d287c",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu, gene.list = genelists.to.plot[[l]], name = l, \n",
    "                 save.dir = unfilt_dir)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "84bc5d49-f281-4574-afa2-364de0f25faa",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Clipped %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5eb98a3c-a1d1-46b7-84e6-bd98a42cfc78",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "unfilt_dir <- paste0(clipped.dir, 'unfiltered_lists/')\n",
    "dir.create(unfilt_dir, recursive = T)\n",
    "\n",
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu, gene.list = genelists.to.plot[[l]], name = l, \n",
    "                 save.dir = unfilt_dir, clip.upper.pct=30)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c157880-f315-414a-b9e5-799cea46b906",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### Scaled %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f10884a-e731-44be-bfcf-8ed7facd0455",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "unfilt_dir <- paste0(scaled.dir, 'unfiltered_lists/')\n",
    "dir.create(unfilt_dir, recursive = T)\n",
    "\n",
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu, gene.list = genelists.to.plot[[l]], name = l, \n",
    "                 pct_scale_by = c('features.plot', 'feature.groups'),\n",
    "                 save.dir = unfilt_dir)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2d52b57-31fe-43d7-b068-fb8316178ed9",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Unfiltered lists without donor 384 (`seu.no384`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d58a7e5d-257e-4f33-8c97-ab0daf6a2ce3",
   "metadata": {},
   "source": [
    "#### Unmodified %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "91f0fb01-77c2-4fd0-97fb-46da9faa822f",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "unfilt.no384_dir <- paste0(original.dir, 'unfiltered_lists.no_384/')\n",
    "dir.create(unfilt.no384_dir, recursive = T)\n",
    "\n",
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu.no384, gene.list = genelists.to.plot[[l]], name = l, \n",
    "                 save.dir = unfilt.no384_dir)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "33400e5c-e1d0-4886-8083-cfe4035731b6",
   "metadata": {},
   "source": [
    "#### Clipped %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85c1aca2-036b-45fd-9f41-02d48694b972",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "unfilt.no384_dir <- paste0(clipped.dir, 'unfiltered_lists.no_384/')\n",
    "dir.create(unfilt.no384_dir, recursive = T)\n",
    "\n",
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu.no384, gene.list = genelists.to.plot[[l]], name = l, \n",
    "                 save.dir = unfilt.no384_dir, clip.upper.pct=30)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a71c830-224c-49d4-bab4-8e398714aaa0",
   "metadata": {},
   "source": [
    "#### Scaled %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebfbecb1-18dd-4900-952d-8fe660fd6acd",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "unfilt.no384_dir <- paste0(scaled.dir, 'unfiltered_lists.no_384/')\n",
    "dir.create(unfilt.no384_dir, recursive = T)\n",
    "\n",
    "\n",
    "# without clustering\n",
    "for (l in names(genelists.to.plot)){\n",
    "    message(l)\n",
    "    dotplot.save(seu = seu.no384, gene.list = genelists.to.plot[[l]], name = l, \n",
    "                 pct_scale_by = c('features.plot', 'feature.groups'),\n",
    "                 save.dir = unfilt.no384_dir)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af614e81-5e8a-446b-a31e-9a255536b723",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Filtered lists by markers **(not needed)**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9702a7b0-4308-434a-9411-1c4a1ca1d615",
   "metadata": {},
   "source": [
    "#### Unmodified %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1c9e44c-c5db-4a94-8540-a67317c348b4",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "filter_dir <- paste0(original.dir, 'filtered_with_markers/')\n",
    "dir.create(filter_dir, recursive=T)\n",
    "\n",
    "for (l in names(genelists.to.plot)){\n",
    "# for (l in c('Imprinted genes')){\n",
    "    \n",
    "    message(l)\n",
    "    \n",
    "    \n",
    "    for (filt in av.filters){\n",
    "        \n",
    "        print(filt)\n",
    "        list.genes <- genelists.to.plot[[l]]\n",
    "        \n",
    "        \n",
    "        # Seurat markers\n",
    "        valid.genes <- markers.all[markers.all[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        dotplot.save(seu = seu, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir, 'sc_per_celltype_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "        \n",
    "        \n",
    "        valid.genes <- markers.rge[markers.rge[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "\n",
    "        # Dotplot\n",
    "        dotplot.save(seu = seu, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir, 'pseudobulk_rge_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "                \n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0ac86eb-c1fe-4f86-9f7c-bfeb90b30c03",
   "metadata": {},
   "source": [
    "#### Clipped %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "946f15da-c040-49d2-983a-60251c3b9db7",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "filter_dir <- paste0(clipped.dir, 'filtered_with_markers/')\n",
    "dir.create(filter_dir, recursive=T)\n",
    "\n",
    "for (l in names(genelists.to.plot)){\n",
    "    \n",
    "    message(l)\n",
    "    \n",
    "    \n",
    "    for (filt in av.filters){\n",
    "        \n",
    "        print(filt)\n",
    "        list.genes <- genelists.to.plot[[l]]\n",
    "        \n",
    "        \n",
    "        # Seurat markers\n",
    "        valid.genes <- markers.all[markers.all[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "\n",
    "        dotplot.save(seu = seu, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir, 'sc_per_celltype_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]], clip.upper.pct=30)\n",
    "        \n",
    "        \n",
    "        valid.genes <- markers.rge[markers.rge[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        # Dotplot\n",
    "        dotplot.save(seu = seu, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir, 'pseudobulk_rge_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]], clip.upper.pct=30)\n",
    "                \n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "10bb0a53-ee7a-41e8-9bc0-f9c02e29fea7",
   "metadata": {},
   "source": [
    "#### Scaled %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddf50474-909f-40c3-bfce-8b2ab2f93dce",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "filter_dir <- paste0(scaled.dir, 'filtered_with_markers/')\n",
    "dir.create(filter_dir, recursive=T)\n",
    "\n",
    "for (l in names(genelists.to.plot)){\n",
    "    \n",
    "    message(l)\n",
    "    \n",
    "    \n",
    "    for (filt in av.filters){\n",
    "        \n",
    "        print(filt)\n",
    "        list.genes <- genelists.to.plot[[l]]\n",
    "        \n",
    "        \n",
    "        # Seurat markers\n",
    "        valid.genes <- markers.all[markers.all[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        dotplot.save(seu = seu, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir, 'sc_per_celltype_markers/', filt, '/'),\n",
    "                     pct_scale_by = c('features.plot', 'feature.groups'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "        \n",
    "        \n",
    "        valid.genes <- markers.rge[markers.rge[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        # Dotplot\n",
    "        dotplot.save(seu = seu, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir, 'pseudobulk_rge_markers/', filt, '/'),\n",
    "                     pct_scale_by = c('features.plot', 'feature.groups'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "                \n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af47a75f-5296-4378-a06f-9fc4b4481ce3",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "### Filtered lists by markers without 384 (`seu.no384`) **(not needed)**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "302170cf-7d7a-4e3f-88b0-f84fac1d7c4f",
   "metadata": {},
   "source": [
    "#### Unmodified %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ad1877dc-80c3-4065-b83f-172e30b7a41d",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "filter_dir.no384 <- paste0(original.dir, 'filtered_with_markers.no_384/')\n",
    "dir.create(filter_dir.no384, recursive=T)\n",
    "\n",
    "for (l in names(genelists.to.plot)){\n",
    "    \n",
    "    message(l)\n",
    "    \n",
    "    \n",
    "    for (filt in av.filters){\n",
    "        \n",
    "        print(filt)\n",
    "        list.genes <- genelists.to.plot[[l]]\n",
    "        \n",
    "        \n",
    "        # Seurat markers\n",
    "        valid.genes <- markers.all[markers.all[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        dotplot.save(seu = seu.no384, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir.no384, 'sc_per_celltype_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "        \n",
    "        \n",
    "        valid.genes <- markers.rge[markers.rge[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        # Dotplot\n",
    "        dotplot.save(seu = seu.no384, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir.no384, 'pseudobulk_rge_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "                \n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c690709-1a02-479c-81ce-6c00add3d0f6",
   "metadata": {},
   "source": [
    "#### Clipped %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2521a5fc-8b3e-4443-a767-5351911e7d93",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "filter_dir.no384 <- paste0(clipped.dir, 'filtered_with_markers.no_384/')\n",
    "dir.create(filter_dir.no384, recursive=T)\n",
    "\n",
    "for (l in names(genelists.to.plot)){\n",
    "    \n",
    "    message(l)\n",
    "    \n",
    "    \n",
    "    for (filt in av.filters){\n",
    "        \n",
    "        print(filt)\n",
    "        list.genes <- genelists.to.plot[[l]]\n",
    "        \n",
    "        \n",
    "        # Seurat markers\n",
    "        valid.genes <- markers.all[markers.all[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        dotplot.save(seu = seu.no384, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir.no384, 'sc_per_celltype_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]], clip.upper.pct=30)\n",
    "        \n",
    "        \n",
    "        valid.genes <- markers.rge[markers.rge[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        # Dotplot\n",
    "        dotplot.save(seu = seu.no384, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir.no384, 'pseudobulk_rge_markers/', filt, '/'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]], clip.upper.pct=30)\n",
    "                \n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5039d26c-3963-412e-a315-ab8b31a4e172",
   "metadata": {},
   "source": [
    "#### Scaled %"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3558ebae-bf74-41a3-838b-181f4605f0c9",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [],
   "source": [
    "filter_dir.no384 <- paste0(scaled.dir, 'filtered_with_markers.no_384/')\n",
    "dir.create(filter_dir.no384, recursive=T)\n",
    "\n",
    "for (l in names(genelists.to.plot)){\n",
    "    \n",
    "    message(l)\n",
    "    \n",
    "    \n",
    "    for (filt in av.filters){\n",
    "        \n",
    "        print(filt)\n",
    "        list.genes <- genelists.to.plot[[l]]\n",
    "        \n",
    "        \n",
    "        # Seurat markers\n",
    "        valid.genes <- markers.all[markers.all[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        dotplot.save(seu = seu.no384, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir.no384, 'sc_per_celltype_markers/', filt, '/'),\n",
    "                     pct_scale_by = c('features.plot', 'feature.groups'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "        \n",
    "        \n",
    "        valid.genes <- markers.rge[markers.rge[,filt],'gene']%>%unique()\n",
    "        valid.genes <- lapply(list.genes, function(x){x[x%in%valid.genes]})\n",
    "        valid.genes <- valid.genes[sapply(valid.genes, length)!=0]\n",
    "        \n",
    "        # Dotplot\n",
    "        dotplot.save(seu = seu.no384, gene.list = valid.genes, name = l, \n",
    "                     save.dir = paste0(filter_dir.no384, 'pseudobulk_rge_markers/', filt, '/'),\n",
    "                     pct_scale_by = c('features.plot', 'feature.groups'),\n",
    "                     title='Genes that are DE between ASD and control samples',\n",
    "                     subtitle=exp.filt.opts[[filt]])\n",
    "                \n",
    "    }\n",
    "    \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ea72692-452c-4c44-ad6a-067de4df6ffe",
   "metadata": {},
   "source": [
    "---"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Portfolio (R)",
   "language": "R",
   "name": "portfolio-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.1"
  },
  "toc-showcode": false
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
